---
title: "Genetic Association Analysis with R (II)"
output:
  bookdown::html_document2:
    toc: true
    toc_float: true
    number_sections: true
fontsize: 11pt
bibliography: '`r system.file("REFERENCES.bib", package="gaawr2")`'
csl: nature-genetics.csl
pkgdown:
  as_is: true
vignette: >
  %\VignetteIndexEntry{gaawr2}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include=FALSE}
set.seed(0)
knitr::opts_chunk$set(
  out.extra = 'style="display:block; margin: auto"',
  fig.align = "center",
  fig.path = "gaawr2/",
  collapse = TRUE,
  comment = "#>",
  dev = "png")
```

```{r, echo=FALSE, message=FALSE, warning=FALSE}
pkgs <- c("EnsDb.Hsapiens.v75","GMMAT","HardyWeinberg","MCMCglmm","biomaRt","gap","haplo.stats","pedigreemm",
           "dplyr","ggplot2","httr","jsonlite","knitr","powerEQTL")
for (p in pkgs) if (length(grep(paste("^package:", p, "$", sep=""), search())) == 0) {
    if (!requireNamespace(p)) warning(paste0("This vignette needs package `", p, "'; please install"))
}
invisible(suppressMessages(lapply(pkgs, require, character.only = TRUE)))
```

This package intends to gather information, meta-data and relevant scripts in genetic association analysis as a companion to a Henry-Stewart talk.

The package is adapted from earlier package development work, particularly pQTLdata.

# Hello, welcome to gaawr2!

We start this session with several ways of printting a welcome message.

## R

The usual and the simplest way is to execte code from the R command line prompt,

```{r welcome}
print("Hello, welcome to gaawr2!")
```

As we are now in a big data era, one might works from a Linux terminal,

## Linux --> R

```bash
export message="Hello, welcome to gaawr2!"
Rscript -e 'message=Sys.getenv("message");print(message)'
```

so the message is wrapped in an environment variable called `message` and grabbed into R.

## Linux --> R --> Linux

This is somewhat unusal but I found extremely useful, notably in its use of R as part of a chain of commands, connected through the pipe
operator (`|`) and in our case there are more than one of them therefore split into many lines while informing the Linux they are in the
same chain, i.e., (`\`),

```bash
echo Hello, welcome to gaawr2 | \
Rscript -e '
   message <- scan("stdin", what = "", sep = "\n", quiet = TRUE);
   write.table(message, col.names = FALSE, row.names = FALSE, quote = FALSE)
' | \
cat
```

Later, we will use some more useful examples on this.

Note that R provides its own batch command to run an R script, i.e., `R CMD BATCH hello.R` assuming our message is contained `hello.R`.

# Operators and packages

The operators are intuitive in many ways, but we would like to highlight two types of them,

- the scope operator (`::`) is useful since user executes command from a particular package without loading it, which is usually faster.
- the native (`|>`) and contributed (`%>%`) pipe operators which enable a chained of operations, the latter popularized from R **magrittr** and **dplyr** packages

```{r ops, fig.height=6, fig.with=8}
sumstats <- mtcars %>%
  dplyr::filter(mpg > 20) %>%
  dplyr::group_by(cyl) %>%
  dplyr::summarize(avg_hp = mean(hp), avg_wt = mean(wt))
knitr::kable(sumstats,caption="")
ggplot2::ggplot(sumstats, ggplot2::aes(x = factor(cyl), y = avg_hp, fill = factor(cyl))) +
ggplot2::geom_bar(stat = "identity") +
ggplot2::labs(
    title = "Average Horsepower by Cylinder Count (mpg > 20)",
    x = "Number of Cylinders",
    y = "Average Horsepower"
  ) +
ggplot2::theme_minimal() +
ggplot2::scale_fill_brewer(palette = "Set2", name = "Cylinders")
```

Two most often used approaches to make R packages available is through

- `install.packages()` which is a standard way to install from the Comprehensive R Archive Network (CRAN) hosting many user-contributed packages.
- `BiocManager::install() which is an updated command to install package from the Bioconductor project.`

# Annotations

## EnsDb.Hsapiens.v75

```{r ensdb, messages=FALSE}
ensembldb::metadata(EnsDb.Hsapiens.v75)
genes <- ensembldb::genes(EnsDb.Hsapiens.v75)
head(genes)
transcripts_data <- ensembldb::transcripts(EnsDb.Hsapiens.v75)
head(transcripts_data)
```
One can also use `exons_data <- ensembldb::exons(EnsDb.Hsapiens.v75);head(exons_data)` but it is skipped for being considerably longer.

## biomaRt

```r
biomaRt::listMarts()
mart <- biomaRt::useMart("ENSEMBL_MART_FUNCGEN")
biomaRt::listDatasets(mart)
mart <- biomaRt::useMart("ensembl")
biomaRt::listDatasets(mart)
ensembl <- biomaRt::useMart(biomart="ensembl", dataset="hsapiens_gene_ensembl", host="grch37.ensembl.org", path="/biomart/martservice")
attr <- biomaRt::listAttributes(ensembl)
attr_select <- c('ensembl_gene_id', 'chromosome_name', 'start_position', 'end_position', 'description', 'hgnc_symbol',
                 'transcription_start_site')
gene <- biomaRt::getBM(attributes = attr_select, mart = ensembl)
filter <- biomaRt::listFilters(ensembl)
biomaRt::searchFilters(mart = ensembl, pattern = "gene")
```

## Experimental Factor Ontology (EFO)

The ontology of traits/disease is formally available as this @malone10.

```r
library(ontologyIndex)
id <- function(ontology)
{
  inflammatory <- grep(ontology$name,pattern="inflammatory")
  immune <- grep(ontology$name,pattern="immune")
  inf <- union(inflammatory,immune)
  list(id=ontology$id[inf],name=ontology$name[inf])
}
# GO
data(go)
goidname <- id(go)
# EFO
file <- "efo.obo" # efo-3.26.0
get_relation_names(file)
efo <- get_ontology(file, extract_tags="everything")
length(efo) # 89
length(efo$id) # 27962
efoidname <- id(efo)
diseases <- get_descendants(efo,"EFO:0000408")
efo_0000540 <- get_descendants(efo,"EFO:0000540")
efo_0000540name <- efo$name[efo_0000540]
isd <- data.frame(efo_0000540,efo_0000540name)
save(efo,diseases,isd,efoidname,goidname, file="efo.rda")
write.table(isd,file="efo_0000540.csv",col.names=FALSE,row.names=FALSE,sep=",")
library(ontologyPlot)
onto_plot(efo,efo_0000540)
```

## rentrez

```r
library(rentrez)
entrez_dbs()
entrez_db_links("pubmed")
pubmed_fields <- entrez_db_searchable("pubmed")
# set_entrez_key("")
Sys.getenv("ENTREZ_KEY")
term <- "pQTLs OR (protein AND quantitative AND trait AND loci) AND human [MH] AND (plasma OR Serum)"
r <- entrez_search(db="pubmed",term=term,use_history=TRUE)
class(r)
names(r)
with(r,web_history)
unlink(paste("pubmed",c("fetch","summary"),sep="."))
fields <- c("uid", "pubdate", "sortfirstauthor", "title", "source", "volume", "pages")
for(i in seq(1,with(r,count),50))
{
  cat(i+49, "records downloaded\r")
  f <- entrez_fetch(db="pubmed", web_history=with(r,web_history), rettype="text", retmax=50, retstart=i)
  write.table(f, col.names=FALSE, row.names=FALSE, file="pubmed.fetch", append=TRUE)
  s <- entrez_summary(db="pubmed", web_history=with(r,web_history), rettype="text", retmax=50, retstart=i)
  e <- extract_from_esummary(s, fields)
  write.table(t(e), col.names=FALSE, row.names=FALSE, file="pubmed.summary", append=TRUE, sep="\t")
}
id <- 600807
upload <- entrez_post(db="omim", id=id)
asthma_variants <- entrez_link(dbfrom="omim", db="clinvar", cmd="neighbor_history", web_history=upload)
asthma_variants
snp_links <- entrez_link(dbfrom="clinvar", db="snp", web_history=asthma_variants$web_histories$omim_clinvar, cmd="neighbor_history")
all_links <- entrez_link(dbfrom='pubmed', id=id, db='all')
```

# Data analysis

Topics vary from those heavily featured in candidate gene studies, such as Hardy-Weinberg equilirium (HWE), to their counterparts in
Genomewide association study (GWAS) such as various types of association statistics, QQ/Manhattan/local association plots.

There has been a lot of interests in machine learning (ML), artificial language (AI), including deep learning (DL), just to add one more
acronym, the bulk of which is also readily available.

## HardyWeinberg

We set to run through the package for HWE. Three data sources are used: MN blood group in the documentation,
a chromosome X SNP and a HLA/DQR,

```{r hwe, fig.cap="SNP ternary plot", fig.height=15.07, fig.width=15.6, messages=FALSE}
options(width=200)
# MN blood group
SNP <- c(MM = 298, MN = 489, NN = 213)
HardyWeinberg::maf(SNP)
HardyWeinberg::HWTernaryPlot(SNP,region=0,hwcurve=FALSE,grid=TRUE,markercol="blue")
HardyWeinberg::HWChisq(SNP, cc = 0, verbose = TRUE)
# Chromosome X
xSNP <- c(A=10, B=20, AA=30, AB=20, BB=10)
HardyWeinberg::HWChisq(xSNP,cc=0,x.linked=TRUE,verbose=TRUE)
# HLA/DQR
library(gap)
DQR <- hla[,3:4]
a1 <- DQR[1]
a2 <- DQR[2]
GenotypeCounts <- HardyWeinberg::AllelesToTriangular(a1,a2)
knitr::kable(GenotypeCounts,caption="Genotype distribution of DQR")
HardyWeinberg::HWPerm.mult(GenotypeCounts,nperm=1000)
HardyWeinberg::HWStr(hla[,3:4],test="permutation",nperm=1000)
```

## SNPassoc

The package implements procedures which are appropriate for candidate gene association analysis, under a variety of genetic models.

## haplo.stats

This package considers haplotype estimation using EM-algorithms and genetic association under a generalized linear model (GLM).

## pedigreemm

kinship2, especially with its `lmekin`.

## GWAS

This includes association modeling, annotation. Graphics, e.g., `inst/scripts/lz.R`.

The following packages are notable, `GWASExactHW`, `GMMAT`, `GWAStools`, `SAIGE`, `qqman`.

## ML & AI

For ML, it glmnet is worthy of note. For AI, some pedagogic implementation such as `nnet` which has long been in R's list of recommended
packages , `neuralnet`, `RSNNS` which is ported from the standalone package, as well as production packages (parallel developments to
Python) include `tensorflow` (TensorFlow), `torch` (PyTorch), and `keras` (Keras).

## Miscellaneous topics

- Parallel computing.
- Data. `SeqArray` format GDS from PLINK, whose code for handling binary encoded data (bed) are widely adopted. Customized packages for databases include RSQLite, RMySQL.
- Graphics, the list includes `Rgraphviz` and `magick` for the the extremely popular `graphviz` and `ImageMagick`.
- Omics. Proteomics such as `MsCoreUtils`, `MSStats`, `scp`. An overview on the topic is found from the relevant paper @suhre20.
- post-GWAS.


### powerEQTL

```{r eqtl, fig.cap="Power Estimation for eQTL Studies of 240 SNPs", fig.height=6, fig.width=8, messages=FALSE}
n.designs <- 6
designs <- 1:n.designs
N <- 50 * designs
n.grids <- 100
index <- 1:n.grids
grids <- index / n.grids
MAF <- seq(0.005, n.grids/2, by=0.5) / n.grids
plot(MAF,grids,type="n",ylab="Power")
mtext(expression(paste("(",alpha," = 0.05)")),1,line=4.5)
colors <- grDevices::hcl.colors(n.designs)
for (design in designs)
{
  power.SLR <- rep(NA,n.grids)
  for (j in index) power.SLR[j] <- powerEQTL::powerEQTL.SLR(MAF = MAF[j], FWER = 0.05, nTests = 240, slope = 0.13,
                                                            n = N[design], sigma.y = 0.13)
  lines(MAF,power.SLR,col=colors[design])
}
legend("bottomright", inset=.02, title="Sample size (N)", paste(N), col=colors, horiz=FALSE, cex=0.8, lty=designs)
```

### OpenTargets

```{r, messages=FALSE}
gene_id <- "ENSG00000164308"
query_string = "
  query target($ensemblId: String!){
    target(ensemblId: $ensemblId){
      id
      approvedSymbol
      biotype
      geneticConstraint {
        constraintType
        exp
        obs
        score
        oe
        oeLower
        oeUpper
      }
      tractability {
        label
        modality
        value
      }
    }
  }
"
base_url <- "https://api.platform.opentargets.org/api/v4/graphql"
variables <- list("ensemblId" = gene_id)
post_body <- list(query = query_string, variables = variables)
r <- httr::POST(url=base_url, body=post_body, encode='json')
data <- iconv(r, "latin1", "ASCII")
content <- jsonlite::fromJSON(data)
knitr::kable(content, caption="Query of ERAP2 from OpenTargets")
```

# gaawr2

While created as a showcase of modern package development, like other R packages it includes data examples, customized functions,
documentation and featured articles.

The relevant scripts are with `inst/scripts` directory in the source package. Briefly,

- `gaawr2.R` creates the package in R.
- `github.sh` creates `gaawr2` at GitHub from the command line.
- `pkgdown.sh` makes a pkgdown-style package and this vignette is set to be processed with the `bookdown` package.
- `docs.sh` adds, commits and pushes files to GitHub.

Note that for creation of the GitHub repository, an SSH key is assumed in place. In order for `pkgdown.sh` to function well, all
required files such as `nature-genetics.csl` need to be available.

A GitHub login is still necessary to enable web pages, so that this can be accessed as <https://jinghuazhao.github.io/gaawr2/>.
Upon use `pkgdown`, an article can be seen from the menu item `Articles`.

We carry on adding files such as `NEWS.md` and `_pkgdown.yml`. In line with various analyses we have covered, their associate packages
are also added to the suggested list of packages in DESCRIPTION:

```
Suggests:
    HardyWeinberg,
    haplo.stats,
    SNPassoc,
    locuszoomr,
    pedigreemm,
    GMMAT,
    SAIGE,
    BLR,
    BGLR,
    MCMCglmm
```

# References
