<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Genetic Association Analysis with R (II) • gaawr2</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Genetic Association Analysis with R (II)">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gaawr2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fa-home"></span></a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><h6 class="dropdown-header" data-toc-skip>Vignettes:</h6></li>
    <li><a class="dropdown-item" href="../articles/gaawr2.html">Genetic Association Analysis with R (II)</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Articles:</h6></li>
    <li><a class="dropdown-item" href="../articles/web.html">Web facilities</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/gaawr2.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/jinghuazhao/gaawr2"><span class="fa fab fa-github"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://jinghuazhao.github.io/R/"><span class="fa fab fa-r-project"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">


<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Genetic Association Analysis with R (II)</h1>
            
      

      <div class="d-none name"><code>gaawr2.Rmd</code></div>
    </div>

    
    
<p>This package intends to gather information, meta-data and relevant scripts in genetic association analysis as a companion to a Henry-Stewart talk.</p>
<p>The package is adapted from earlier package development work, particularly pQTLdata.</p>
<div class="section level2" number="1">
<h2 id="hello-welcome-to-gaawr2">
<span class="header-section-number">1</span> Hello, welcome to gaawr2!<a class="anchor" aria-label="anchor" href="#hello-welcome-to-gaawr2"></a>
</h2>
<p>We start this session with several ways of printting a welcome message.</p>
<div class="section level3" number="1.1">
<h3 id="r">
<span class="header-section-number">1.1</span> R<a class="anchor" aria-label="anchor" href="#r"></a>
</h3>
<p>The usual and the simplest way is to execte code from the R command line prompt,</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="st">"Hello, welcome to gaawr2!"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Hello, welcome to gaawr2!"</span></span></code></pre></div>
<p>As we are now in a big data era, one might works from a Linux terminal,</p>
</div>
<div class="section level3" number="1.2">
<h3 id="linux-r">
<span class="header-section-number">1.2</span> Linux –&gt; R<a class="anchor" aria-label="anchor" href="#linux-r"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">message</span><span class="op">=</span><span class="st">"Hello, welcome to gaawr2!"</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="ex">Rscript</span> <span class="at">-e</span> <span class="st">'message=Sys.getenv("message");print(message)'</span></span></code></pre></div>
<p>so the message is wrapped in an environment variable called <code>message</code> and grabbed into R.</p>
</div>
<div class="section level3" number="1.3">
<h3 id="linux-r-linux">
<span class="header-section-number">1.3</span> Linux –&gt; R –&gt; Linux<a class="anchor" aria-label="anchor" href="#linux-r-linux"></a>
</h3>
<p>This is somewhat unusal but I found extremely useful, notably in its use of R as part of a chain of commands, connected through the pipe
operator (<code>|</code>) and in our case there are more than one of them therefore split into many lines while informing the Linux they are in the
same chain, i.e., (<code>\</code>),</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> Hello, welcome to gaawr2 <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="ex">Rscript</span> <span class="at">-e</span> <span class="st">'</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="st">   message &lt;- scan("stdin", what = "", sep = "\n", quiet = TRUE);</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="st">   write.table(message, col.names = FALSE, row.names = FALSE, quote = FALSE)</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="st">'</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span></span></code></pre></div>
<p>Later, we will use some more useful examples on this.</p>
<p>Note that R provides its own batch command to run an R script, i.e., <code>R CMD BATCH hello.R</code> assuming our message is contained <code>hello.R</code>.</p>
</div>
</div>
<div class="section level2" number="2">
<h2 id="operators-and-packages">
<span class="header-section-number">2</span> Operators and packages<a class="anchor" aria-label="anchor" href="#operators-and-packages"></a>
</h2>
<p>The operators are intuitive in many ways, but we would like to highlight two of them,</p>
<ul>
<li>
<code>|&gt;</code> which is a pipe operator enabling many commands piped in the same way as in Linux.</li>
<li>
<code>%&gt;%</code> popularized from R magritter package, essentially another yet more powerful pipe command.</li>
</ul>
<p>Two most often used approaches to make R packages available is through</p>
<ul>
<li>
<code><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages()</a></code> which is a standard way to install from the Comprehensive R Archive Network (CRAN) hosting many user-contributed packages.</li>
<li><code>BiocManager::install() which is an updated command to install package from the Bioconductor project.</code></li>
</ul>
<p>Note the scope operator (<code>::</code>) is useful since user executes command from a particular package without loading it, which is usually faster. With growing popularity of GitHub and also due to its less strict control over format from CRAN, it is customarily used but we will leave this aspect later on.</p>
</div>
<div class="section level2" number="3">
<h2 id="annotation">
<span class="header-section-number">3</span> Annotation<a class="anchor" aria-label="anchor" href="#annotation"></a>
</h2>
<div class="section level3" number="3.1">
<h3 id="ensdb-hsapiens-v75">
<span class="header-section-number">3.1</span> EnsDb.Hsapiens.v75<a class="anchor" aria-label="anchor" href="#ensdb-hsapiens-v75"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ensembldb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">EnsDb.Hsapiens.v75</span><span class="op">)</span></span>
<span><span class="co">#&gt;                  name                               value</span></span>
<span><span class="co">#&gt; 1             Db type                               EnsDb</span></span>
<span><span class="co">#&gt; 2     Type of Gene ID                     Ensembl Gene ID</span></span>
<span><span class="co">#&gt; 3  Supporting package                           ensembldb</span></span>
<span><span class="co">#&gt; 4       Db created by ensembldb package from Bioconductor</span></span>
<span><span class="co">#&gt; 5      script_version                               0.3.0</span></span>
<span><span class="co">#&gt; 6       Creation time            Thu May 18 09:15:45 2017</span></span>
<span><span class="co">#&gt; 7     ensembl_version                                  75</span></span>
<span><span class="co">#&gt; 8        ensembl_host                           localhost</span></span>
<span><span class="co">#&gt; 9            Organism                        homo_sapiens</span></span>
<span><span class="co">#&gt; 10        taxonomy_id                                9606</span></span>
<span><span class="co">#&gt; 11       genome_build                              GRCh37</span></span>
<span><span class="co">#&gt; 12    DBSCHEMAVERSION                                 2.0</span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu">ensembldb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">EnsDb.Hsapiens.v75</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRanges object with 6 ranges and 6 metadata columns:</span></span>
<span><span class="co">#&gt;                   seqnames      ranges strand |         gene_id   gene_name</span></span>
<span><span class="co">#&gt;                      &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;   ENSG00000223972        1 11869-14412      + | ENSG00000223972     DDX11L1</span></span>
<span><span class="co">#&gt;   ENSG00000227232        1 14363-29806      - | ENSG00000227232      WASH7P</span></span>
<span><span class="co">#&gt;   ENSG00000243485        1 29554-31109      + | ENSG00000243485  MIR1302-10</span></span>
<span><span class="co">#&gt;   ENSG00000237613        1 34554-36081      - | ENSG00000237613     FAM138A</span></span>
<span><span class="co">#&gt;   ENSG00000268020        1 52473-54936      + | ENSG00000268020      OR4G4P</span></span>
<span><span class="co">#&gt;   ENSG00000240361        1 62948-63887      + | ENSG00000240361     OR4G11P</span></span>
<span><span class="co">#&gt;                   gene_biotype seq_coord_system      symbol</span></span>
<span><span class="co">#&gt;                    &lt;character&gt;      &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;   ENSG00000223972   pseudogene       chromosome     DDX11L1</span></span>
<span><span class="co">#&gt;   ENSG00000227232   pseudogene       chromosome      WASH7P</span></span>
<span><span class="co">#&gt;   ENSG00000243485      lincRNA       chromosome  MIR1302-10</span></span>
<span><span class="co">#&gt;   ENSG00000237613      lincRNA       chromosome     FAM138A</span></span>
<span><span class="co">#&gt;   ENSG00000268020   pseudogene       chromosome      OR4G4P</span></span>
<span><span class="co">#&gt;   ENSG00000240361   pseudogene       chromosome     OR4G11P</span></span>
<span><span class="co">#&gt;                                            entrezid</span></span>
<span><span class="co">#&gt;                                              &lt;list&gt;</span></span>
<span><span class="co">#&gt;   ENSG00000223972               100287596,100287102</span></span>
<span><span class="co">#&gt;   ENSG00000227232                  100287171,653635</span></span>
<span><span class="co">#&gt;   ENSG00000243485 100422919,100422834,100422831,...</span></span>
<span><span class="co">#&gt;   ENSG00000237613              654835,645520,641702</span></span>
<span><span class="co">#&gt;   ENSG00000268020                              &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   ENSG00000240361                              &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 273 sequences (1 circular) from GRCh37 genome</span></span>
<span><span class="va">transcripts_data</span> <span class="op">&lt;-</span> <span class="fu">ensembldb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">transcripts</a></span><span class="op">(</span><span class="va">EnsDb.Hsapiens.v75</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">transcripts_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRanges object with 6 ranges and 6 metadata columns:</span></span>
<span><span class="co">#&gt;                   seqnames      ranges strand |           tx_id</span></span>
<span><span class="co">#&gt;                      &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;</span></span>
<span><span class="co">#&gt;   ENST00000456328        1 11869-14409      + | ENST00000456328</span></span>
<span><span class="co">#&gt;   ENST00000515242        1 11872-14412      + | ENST00000515242</span></span>
<span><span class="co">#&gt;   ENST00000518655        1 11874-14409      + | ENST00000518655</span></span>
<span><span class="co">#&gt;   ENST00000450305        1 12010-13670      + | ENST00000450305</span></span>
<span><span class="co">#&gt;   ENST00000438504        1 14363-29370      - | ENST00000438504</span></span>
<span><span class="co">#&gt;   ENST00000541675        1 14363-24886      - | ENST00000541675</span></span>
<span><span class="co">#&gt;                               tx_biotype tx_cds_seq_start tx_cds_seq_end</span></span>
<span><span class="co">#&gt;                              &lt;character&gt;        &lt;integer&gt;      &lt;integer&gt;</span></span>
<span><span class="co">#&gt;   ENST00000456328   processed_transcript             &lt;NA&gt;           &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   ENST00000515242 transcribed_unproces..             &lt;NA&gt;           &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   ENST00000518655 transcribed_unproces..             &lt;NA&gt;           &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   ENST00000450305 transcribed_unproces..             &lt;NA&gt;           &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   ENST00000438504 unprocessed_pseudogene             &lt;NA&gt;           &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   ENST00000541675 unprocessed_pseudogene             &lt;NA&gt;           &lt;NA&gt;</span></span>
<span><span class="co">#&gt;                           gene_id         tx_name</span></span>
<span><span class="co">#&gt;                       &lt;character&gt;     &lt;character&gt;</span></span>
<span><span class="co">#&gt;   ENST00000456328 ENSG00000223972 ENST00000456328</span></span>
<span><span class="co">#&gt;   ENST00000515242 ENSG00000223972 ENST00000515242</span></span>
<span><span class="co">#&gt;   ENST00000518655 ENSG00000223972 ENST00000518655</span></span>
<span><span class="co">#&gt;   ENST00000450305 ENSG00000223972 ENST00000450305</span></span>
<span><span class="co">#&gt;   ENST00000438504 ENSG00000227232 ENST00000438504</span></span>
<span><span class="co">#&gt;   ENST00000541675 ENSG00000227232 ENST00000541675</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 273 sequences (1 circular) from GRCh37 genome</span></span></code></pre></div>
<p>One can also use <code>exons_data &lt;- ensembldb::exons(EnsDb.Hsapiens.v75);head(exons_data)</code> but it is skipped for being considerably longer.</p>
</div>
<div class="section level3" number="3.2">
<h3 id="biomart">
<span class="header-section-number">3.2</span> biomaRt<a class="anchor" aria-label="anchor" href="#biomart"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listMarts.html" class="external-link">listMarts</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">mart</span> <span class="op">&lt;-</span> <span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useMart.html" class="external-link">useMart</a></span><span class="op">(</span><span class="st">"ENSEMBL_MART_FUNCGEN"</span><span class="op">)</span></span>
<span><span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listDatasets.html" class="external-link">listDatasets</a></span><span class="op">(</span><span class="va">mart</span><span class="op">)</span></span>
<span><span class="va">mart</span> <span class="op">&lt;-</span> <span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useMart.html" class="external-link">useMart</a></span><span class="op">(</span><span class="st">"ensembl"</span><span class="op">)</span></span>
<span><span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listDatasets.html" class="external-link">listDatasets</a></span><span class="op">(</span><span class="va">mart</span><span class="op">)</span></span>
<span><span class="va">ensembl</span> <span class="op">&lt;-</span> <span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useMart.html" class="external-link">useMart</a></span><span class="op">(</span>biomart<span class="op">=</span><span class="st">"ensembl"</span>, dataset<span class="op">=</span><span class="st">"hsapiens_gene_ensembl"</span>, host<span class="op">=</span><span class="st">"grch37.ensembl.org"</span>, path<span class="op">=</span><span class="st">"/biomart/martservice"</span><span class="op">)</span></span>
<span><span class="va">attr</span> <span class="op">&lt;-</span> <span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listAttributes.html" class="external-link">listAttributes</a></span><span class="op">(</span><span class="va">ensembl</span><span class="op">)</span></span>
<span><span class="va">attr_select</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'ensembl_gene_id'</span>, <span class="st">'chromosome_name'</span>, <span class="st">'start_position'</span>, <span class="st">'end_position'</span>, <span class="st">'description'</span>, <span class="st">'hgnc_symbol'</span>,</span>
<span>                 <span class="st">'transcription_start_site'</span><span class="op">)</span></span>
<span><span class="va">gene</span> <span class="op">&lt;-</span> <span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/getBM.html" class="external-link">getBM</a></span><span class="op">(</span>attributes <span class="op">=</span> <span class="va">attr_select</span>, mart <span class="op">=</span> <span class="va">ensembl</span><span class="op">)</span></span>
<span><span class="va">filter</span> <span class="op">&lt;-</span> <span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listFilters.html" class="external-link">listFilters</a></span><span class="op">(</span><span class="va">ensembl</span><span class="op">)</span></span>
<span><span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listFilters.html" class="external-link">searchFilters</a></span><span class="op">(</span>mart <span class="op">=</span> <span class="va">ensembl</span>, pattern <span class="op">=</span> <span class="st">"gene"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="3.3">
<h3 id="experimental-factor-ontology-efo">
<span class="header-section-number">3.3</span> Experimental Factor Ontology (EFO)<a class="anchor" aria-label="anchor" href="#experimental-factor-ontology-efo"></a>
</h3>
<p>The ontology of traits/disease is formally available as this<span class="citation"><sup>1</sup></span>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ontologyIndex</span><span class="op">)</span></span>
<span><span class="va">id</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">ontology</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="va">inflammatory</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="va">ontology</span><span class="op">$</span><span class="va">name</span>,pattern<span class="op">=</span><span class="st">"inflammatory"</span><span class="op">)</span></span>
<span>  <span class="va">immune</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="va">ontology</span><span class="op">$</span><span class="va">name</span>,pattern<span class="op">=</span><span class="st">"immune"</span><span class="op">)</span></span>
<span>  <span class="va">inf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">union</a></span><span class="op">(</span><span class="va">inflammatory</span>,<span class="va">immune</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>id<span class="op">=</span><span class="va">ontology</span><span class="op">$</span><span class="va">id</span><span class="op">[</span><span class="va">inf</span><span class="op">]</span>,name<span class="op">=</span><span class="va">ontology</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">inf</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co"># GO</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go</span><span class="op">)</span></span>
<span><span class="va">goidname</span> <span class="op">&lt;-</span> <span class="fu">id</span><span class="op">(</span><span class="va">go</span><span class="op">)</span></span>
<span><span class="co"># EFO</span></span>
<span><span class="va">file</span> <span class="op">&lt;-</span> <span class="st">"efo.obo"</span> <span class="co"># efo-3.26.0</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/ontologyIndex/man/get_relation_names.html" class="external-link">get_relation_names</a></span><span class="op">(</span><span class="va">file</span><span class="op">)</span></span>
<span><span class="va">efo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ontologyIndex/man/get_ontology.html" class="external-link">get_ontology</a></span><span class="op">(</span><span class="va">file</span>, extract_tags<span class="op">=</span><span class="st">"everything"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">efo</span><span class="op">)</span> <span class="co"># 89</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">efo</span><span class="op">$</span><span class="va">id</span><span class="op">)</span> <span class="co"># 27962</span></span>
<span><span class="va">efoidname</span> <span class="op">&lt;-</span> <span class="fu">id</span><span class="op">(</span><span class="va">efo</span><span class="op">)</span></span>
<span><span class="va">diseases</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ontologyIndex/man/get_descendants.html" class="external-link">get_descendants</a></span><span class="op">(</span><span class="va">efo</span>,<span class="st">"EFO:0000408"</span><span class="op">)</span></span>
<span><span class="va">efo_0000540</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ontologyIndex/man/get_descendants.html" class="external-link">get_descendants</a></span><span class="op">(</span><span class="va">efo</span>,<span class="st">"EFO:0000540"</span><span class="op">)</span></span>
<span><span class="va">efo_0000540name</span> <span class="op">&lt;-</span> <span class="va">efo</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">efo_0000540</span><span class="op">]</span></span>
<span><span class="va">isd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">efo_0000540</span>,<span class="va">efo_0000540name</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/save.html" class="external-link">save</a></span><span class="op">(</span><span class="va">efo</span>,<span class="va">diseases</span>,<span class="va">isd</span>,<span class="va">efoidname</span>,<span class="va">goidname</span>, file<span class="op">=</span><span class="st">"efo.rda"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">isd</span>,file<span class="op">=</span><span class="st">"efo_0000540.csv"</span>,col.names<span class="op">=</span><span class="cn">FALSE</span>,row.names<span class="op">=</span><span class="cn">FALSE</span>,sep<span class="op">=</span><span class="st">","</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ontologyPlot</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/ontologyPlot/man/onto_plot.html" class="external-link">onto_plot</a></span><span class="op">(</span><span class="va">efo</span>,<span class="va">efo_0000540</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="3.4">
<h3 id="rentrez">
<span class="header-section-number">3.4</span> rentrez<a class="anchor" aria-label="anchor" href="#rentrez"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/rentrez/" class="external-link">rentrez</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_dbs.html" class="external-link">entrez_dbs</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_db_links.html" class="external-link">entrez_db_links</a></span><span class="op">(</span><span class="st">"pubmed"</span><span class="op">)</span></span>
<span><span class="va">pubmed_fields</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_db_searchable.html" class="external-link">entrez_db_searchable</a></span><span class="op">(</span><span class="st">"pubmed"</span><span class="op">)</span></span>
<span><span class="co"># set_entrez_key("")</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html" class="external-link">Sys.getenv</a></span><span class="op">(</span><span class="st">"ENTREZ_KEY"</span><span class="op">)</span></span>
<span><span class="va">term</span> <span class="op">&lt;-</span> <span class="st">"pQTLs OR (protein AND quantitative AND trait AND loci) AND human [MH] AND (plasma OR Serum)"</span></span>
<span><span class="va">r</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_search.html" class="external-link">entrez_search</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"pubmed"</span>,term<span class="op">=</span><span class="va">term</span>,use_history<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">r</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">r</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">r</span>,<span class="va">web_history</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"pubmed"</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"fetch"</span>,<span class="st">"summary"</span><span class="op">)</span>,sep<span class="op">=</span><span class="st">"."</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fields</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"uid"</span>, <span class="st">"pubdate"</span>, <span class="st">"sortfirstauthor"</span>, <span class="st">"title"</span>, <span class="st">"source"</span>, <span class="st">"volume"</span>, <span class="st">"pages"</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">r</span>,<span class="va">count</span><span class="op">)</span>,<span class="fl">50</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="va">i</span><span class="op">+</span><span class="fl">49</span>, <span class="st">"records downloaded\r"</span><span class="op">)</span></span>
<span>  <span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_fetch.html" class="external-link">entrez_fetch</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"pubmed"</span>, web_history<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">r</span>,<span class="va">web_history</span><span class="op">)</span>, rettype<span class="op">=</span><span class="st">"text"</span>, retmax<span class="op">=</span><span class="fl">50</span>, retstart<span class="op">=</span><span class="va">i</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">f</span>, col.names<span class="op">=</span><span class="cn">FALSE</span>, row.names<span class="op">=</span><span class="cn">FALSE</span>, file<span class="op">=</span><span class="st">"pubmed.fetch"</span>, append<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span>  <span class="va">s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_summary.html" class="external-link">entrez_summary</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"pubmed"</span>, web_history<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">r</span>,<span class="va">web_history</span><span class="op">)</span>, rettype<span class="op">=</span><span class="st">"text"</span>, retmax<span class="op">=</span><span class="fl">50</span>, retstart<span class="op">=</span><span class="va">i</span><span class="op">)</span></span>
<span>  <span class="va">e</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/extract_from_esummary.html" class="external-link">extract_from_esummary</a></span><span class="op">(</span><span class="va">s</span>, <span class="va">fields</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">e</span><span class="op">)</span>, col.names<span class="op">=</span><span class="cn">FALSE</span>, row.names<span class="op">=</span><span class="cn">FALSE</span>, file<span class="op">=</span><span class="st">"pubmed.summary"</span>, append<span class="op">=</span><span class="cn">TRUE</span>, sep<span class="op">=</span><span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">id</span> <span class="op">&lt;-</span> <span class="fl">600807</span></span>
<span><span class="va">upload</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_post.html" class="external-link">entrez_post</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"omim"</span>, id<span class="op">=</span><span class="va">id</span><span class="op">)</span></span>
<span><span class="va">asthma_variants</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_link.html" class="external-link">entrez_link</a></span><span class="op">(</span>dbfrom<span class="op">=</span><span class="st">"omim"</span>, db<span class="op">=</span><span class="st">"clinvar"</span>, cmd<span class="op">=</span><span class="st">"neighbor_history"</span>, web_history<span class="op">=</span><span class="va">upload</span><span class="op">)</span></span>
<span><span class="va">asthma_variants</span></span>
<span><span class="va">snp_links</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_link.html" class="external-link">entrez_link</a></span><span class="op">(</span>dbfrom<span class="op">=</span><span class="st">"clinvar"</span>, db<span class="op">=</span><span class="st">"snp"</span>, web_history<span class="op">=</span><span class="va">asthma_variants</span><span class="op">$</span><span class="va">web_histories</span><span class="op">$</span><span class="va">omim_clinvar</span>, cmd<span class="op">=</span><span class="st">"neighbor_history"</span><span class="op">)</span></span>
<span><span class="va">all_links</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_link.html" class="external-link">entrez_link</a></span><span class="op">(</span>dbfrom<span class="op">=</span><span class="st">'pubmed'</span>, id<span class="op">=</span><span class="va">id</span>, db<span class="op">=</span><span class="st">'all'</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2" number="4">
<h2 id="packages-for-genetic-data-analysis">
<span class="header-section-number">4</span> Packages for genetic data analysis<a class="anchor" aria-label="anchor" href="#packages-for-genetic-data-analysis"></a>
</h2>
<div class="section level3" number="4.1">
<h3 id="hardyweinberg">
<span class="header-section-number">4.1</span> HardyWeinberg<a class="anchor" aria-label="anchor" href="#hardyweinberg"></a>
</h3>
<p>We set to run through the package for Hardy-Weinberg equilirium (HWE). Three data sources are used: MN blood group in the documentation,
a chromosome X SNP and a HLA/DQR,</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>width<span class="op">=</span><span class="fl">200</span><span class="op">)</span></span>
<span><span class="co"># MN blood group</span></span>
<span><span class="va">SNP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>MM <span class="op">=</span> <span class="fl">298</span>, MN <span class="op">=</span> <span class="fl">489</span>, NN <span class="op">=</span> <span class="fl">213</span><span class="op">)</span></span>
<span><span class="fu">HardyWeinberg</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HardyWeinberg/man/maf.html" class="external-link">maf</a></span><span class="op">(</span><span class="va">SNP</span><span class="op">)</span></span>
<span><span class="co">#&gt;      N </span></span>
<span><span class="co">#&gt; 0.4575</span></span>
<span><span class="fu">HardyWeinberg</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HardyWeinberg/man/HWTernaryPlot.html" class="external-link">HWTernaryPlot</a></span><span class="op">(</span><span class="va">SNP</span>,region<span class="op">=</span><span class="fl">0</span>,hwcurve<span class="op">=</span><span class="cn">FALSE</span>,grid<span class="op">=</span><span class="cn">TRUE</span>,markercol<span class="op">=</span><span class="st">"blue"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:hwe"></span>
<img src="gaawr2%2Fhwe-1.png" alt="SNP ternary plot" width="1497.6" style="display:block; margin: auto"><p class="caption">
Figure 4.1: SNP ternary plot
</p>
</div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">HardyWeinberg</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HardyWeinberg/man/HWChisq.html" class="external-link">HWChisq</a></span><span class="op">(</span><span class="va">SNP</span>, cc <span class="op">=</span> <span class="fl">0</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Chi-square test for Hardy-Weinberg equilibrium (autosomal)</span></span>
<span><span class="co">#&gt; Chi2 =  0.2214896 DF =  1 p-value =  0.6379073 D =  -3.69375 f =  0.01488253</span></span>
<span><span class="co"># Chromosome X</span></span>
<span><span class="va">xSNP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="fl">10</span>, B<span class="op">=</span><span class="fl">20</span>, AA<span class="op">=</span><span class="fl">30</span>, AB<span class="op">=</span><span class="fl">20</span>, BB<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu">HardyWeinberg</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HardyWeinberg/man/HWChisq.html" class="external-link">HWChisq</a></span><span class="op">(</span><span class="va">xSNP</span>,cc<span class="op">=</span><span class="fl">0</span>,x.linked<span class="op">=</span><span class="cn">TRUE</span>,verbose<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Chi-square test for Hardy-Weinberg equilibrium (X-chromosomal)</span></span>
<span><span class="co">#&gt; Chi2 =  14.86111 DF = 2 p-value =  0.0005928581 D =  NA f =  0.25</span></span>
<span><span class="co"># HLA/DQR</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jinghuazhao/R" class="external-link">gap</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: gap.datasets</span></span>
<span><span class="co">#&gt; gap version 1.6</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'gap'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:HardyWeinberg':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     qqunif</span></span>
<span><span class="va">DQR</span> <span class="op">&lt;-</span> <span class="va">hla</span><span class="op">[</span>,<span class="fl">3</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span>
<span><span class="va">a1</span> <span class="op">&lt;-</span> <span class="va">DQR</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">a2</span> <span class="op">&lt;-</span> <span class="va">DQR</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="va">GenotypeCounts</span> <span class="op">&lt;-</span> <span class="fu">HardyWeinberg</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HardyWeinberg/man/AllelesToTriangular.html" class="external-link">AllelesToTriangular</a></span><span class="op">(</span><span class="va">a1</span>,<span class="va">a2</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">GenotypeCounts</span>,caption<span class="op">=</span><span class="st">"Genotype distribution of DQR"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>
<span id="tab:hwe">Table 4.1: </span>Genotype distribution of DQR</caption>
<colgroup>
<col width="4%">
<col width="3%">
<col width="3%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="3%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">A0</th>
<th align="right">A1</th>
<th align="right">A10</th>
<th align="right">A11</th>
<th align="right">A12</th>
<th align="right">A13</th>
<th align="right">A14</th>
<th align="right">A15</th>
<th align="right">A16</th>
<th align="right">A17</th>
<th align="right">A18</th>
<th align="right">A19</th>
<th align="right">A2</th>
<th align="right">A20</th>
<th align="right">A21</th>
<th align="right">A22</th>
<th align="right">A23</th>
<th align="right">A24</th>
<th align="right">A25</th>
<th align="right">A3</th>
<th align="right">A4</th>
<th align="right">A6</th>
<th align="right">A7</th>
<th align="right">A8</th>
<th align="right">A9</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">A0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A1</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A10</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A11</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A12</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A13</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A14</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A15</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A16</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A17</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A18</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A19</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A20</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A21</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A22</td>
<td align="right">0</td>
<td align="right">13</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A23</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A24</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A25</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A3</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A4</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">8</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A6</td>
<td align="right">0</td>
<td align="right">6</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">7</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">11</td>
<td align="right">7</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A7</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A8</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">9</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A9</td>
<td align="right">0</td>
<td align="right">8</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">15</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">7</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">2</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">HardyWeinberg</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HardyWeinberg/man/HWPerm.mult.html" class="external-link">HWPerm.mult</a></span><span class="op">(</span><span class="va">GenotypeCounts</span>,nperm<span class="op">=</span><span class="fl">1000</span><span class="op">)</span></span>
<span><span class="co">#&gt; Permutation test for Hardy-Weinberg equilibrium (autosomal).</span></span>
<span><span class="co">#&gt; 25 alleles detected.</span></span>
<span><span class="co">#&gt; Observed statistic: 1.461325e-84   1000 permutations. p-value: 0.023</span></span>
<span><span class="fu">HardyWeinberg</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HardyWeinberg/man/HWStr.html" class="external-link">HWStr</a></span><span class="op">(</span><span class="va">hla</span><span class="op">[</span>,<span class="fl">3</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>,test<span class="op">=</span><span class="st">"permutation"</span>,nperm<span class="op">=</span><span class="fl">1000</span><span class="op">)</span></span>
<span><span class="co">#&gt; 1 STRs detected.</span></span>
<span><span class="co">#&gt;      STR   N Nt     MinorAF   MajorAF        Ho        He       Hp  pval</span></span>
<span><span class="co">#&gt; 1 DQR.a1 271 25 0.001845018 0.1494465 0.8856089 0.9097643 2.654949 0.026</span></span></code></pre></div>
</div>
<div class="section level3" number="4.2">
<h3 id="snpassoc">
<span class="header-section-number">4.2</span> SNPassoc<a class="anchor" aria-label="anchor" href="#snpassoc"></a>
</h3>
<p>The package implements procedures which are appropriate for candidate gene association analysis, under a variety of genetic models.</p>
</div>
<div class="section level3" number="4.3">
<h3 id="haplo-stats">
<span class="header-section-number">4.3</span> haplo.stats<a class="anchor" aria-label="anchor" href="#haplo-stats"></a>
</h3>
<p>This package considers haplotype estimation using EM-algorithms and genetic association under a generalized linear model (GLM).</p>
</div>
<div class="section level3" number="4.4">
<h3 id="pedigreemm">
<span class="header-section-number">4.4</span> pedigreemm<a class="anchor" aria-label="anchor" href="#pedigreemm"></a>
</h3>
</div>
<div class="section level3" number="4.5">
<h3 id="kinship2">
<span class="header-section-number">4.5</span> kinship2<a class="anchor" aria-label="anchor" href="#kinship2"></a>
</h3>
</div>
<div class="section level3" number="4.6">
<h3 id="genomewide-association-study-gwas">
<span class="header-section-number">4.6</span> Genomewide association study (GWAS)<a class="anchor" aria-label="anchor" href="#genomewide-association-study-gwas"></a>
</h3>
<p>This includes association modeling, annotation. Graphics, e.g., <code>inst/scripts/lz.R</code>.</p>
</div>
<div class="section level3" number="4.7">
<h3 id="proteogenomics">
<span class="header-section-number">4.7</span> Proteogenomics<a class="anchor" aria-label="anchor" href="#proteogenomics"></a>
</h3>
<p>An overview on the topic is found from the relevant paper<span class="citation"><sup>2</sup></span>.</p>
</div>
<div class="section level3" number="4.8">
<h3 id="opentargets">
<span class="header-section-number">4.8</span> OpenTargets<a class="anchor" aria-label="anchor" href="#opentargets"></a>
</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_id</span> <span class="op">&lt;-</span> <span class="st">"ENSG00000164308"</span></span>
<span><span class="va">query_string</span> <span class="op">=</span> <span class="st">"</span></span>
<span><span class="st">  query target($ensemblId: String!){</span></span>
<span><span class="st">    target(ensemblId: $ensemblId){</span></span>
<span><span class="st">      id</span></span>
<span><span class="st">      approvedSymbol</span></span>
<span><span class="st">      biotype</span></span>
<span><span class="st">      geneticConstraint {</span></span>
<span><span class="st">        constraintType</span></span>
<span><span class="st">        exp</span></span>
<span><span class="st">        obs</span></span>
<span><span class="st">        score</span></span>
<span><span class="st">        oe</span></span>
<span><span class="st">        oeLower</span></span>
<span><span class="st">        oeUpper</span></span>
<span><span class="st">      }</span></span>
<span><span class="st">      tractability {</span></span>
<span><span class="st">        label</span></span>
<span><span class="st">        modality</span></span>
<span><span class="st">        value</span></span>
<span><span class="st">      }</span></span>
<span><span class="st">    }</span></span>
<span><span class="st">  }</span></span>
<span><span class="st">"</span></span>
<span><span class="va">base_url</span> <span class="op">&lt;-</span> <span class="st">"https://api.platform.opentargets.org/api/v4/graphql"</span></span>
<span><span class="va">variables</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"ensemblId"</span> <span class="op">=</span> <span class="va">gene_id</span><span class="op">)</span></span>
<span><span class="va">post_body</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">query_string</span>, variables <span class="op">=</span> <span class="va">variables</span><span class="op">)</span></span>
<span><span class="va">r</span> <span class="op">&lt;-</span> <span class="fu">httr</span><span class="fu">::</span><span class="fu"><a href="https://httr.r-lib.org/reference/POST.html" class="external-link">POST</a></span><span class="op">(</span>url<span class="op">=</span><span class="va">base_url</span>, body<span class="op">=</span><span class="va">post_body</span>, encode<span class="op">=</span><span class="st">'json'</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/iconv.html" class="external-link">iconv</a></span><span class="op">(</span><span class="va">r</span>, <span class="st">"latin1"</span>, <span class="st">"ASCII"</span><span class="op">)</span></span>
<span><span class="co">#&gt; No encoding supplied: defaulting to UTF-8.</span></span>
<span><span class="va">content</span> <span class="op">&lt;-</span> <span class="fu">jsonlite</span><span class="fu">::</span><span class="fu"><a href="https://jeroen.r-universe.dev/jsonlite/reference/fromJSON.html" class="external-link">fromJSON</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">content</span>, caption<span class="op">=</span><span class="st">"Query of ERAP2 from OpenTargets"</span><span class="op">)</span></span></code></pre></div>
<table class="table kable_wrapper">
<caption>
<span id="tab:unnamed-chunk-2">Table 4.2: </span>Query of ERAP2 from OpenTargets
</caption>
<tbody><tr>
<td>
<table class="table kable_wrapper"><tbody><tr>
<td>
<table class="table kable_wrapper"><tbody><tr>
<td>
<table class="table">
<thead><tr class="header">
<th align="left">x</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">ENSG00000164308</td>
</tr></tbody>
</table>
</td>
<td>
<table class="table">
<thead><tr class="header">
<th align="left">x</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">ERAP2</td>
</tr></tbody>
</table>
</td>
<td>
<table class="table">
<thead><tr class="header">
<th align="left">x</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">protein_coding</td>
</tr></tbody>
</table>
</td>
<td>
<table class="table">
<thead><tr class="header">
<th align="left">constraintType</th>
<th align="right">exp</th>
<th align="right">obs</th>
<th align="right">score</th>
<th align="right">oe</th>
<th align="right">oeLower</th>
<th align="right">oeUpper</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">syn</td>
<td align="right">185.590</td>
<td align="right">183</td>
<td align="right">0.14966</td>
<td align="right">0.98603</td>
<td align="right">0.873</td>
<td align="right">1.115</td>
</tr>
<tr class="even">
<td align="left">mis</td>
<td align="right">496.520</td>
<td align="right">501</td>
<td align="right">-0.07147</td>
<td align="right">1.00900</td>
<td align="right">0.937</td>
<td align="right">1.086</td>
</tr>
<tr class="odd">
<td align="left">lof</td>
<td align="right">48.461</td>
<td align="right">45</td>
<td align="right">0.00000</td>
<td align="right">0.92859</td>
<td align="right">0.730</td>
<td align="right">1.191</td>
</tr>
</tbody>
</table>
</td>
<td>
<table class="table">
<thead><tr class="header">
<th align="left">label</th>
<th align="left">modality</th>
<th align="left">value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Approved Drug</td>
<td align="left">SM</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">Advanced Clinical</td>
<td align="left">SM</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">Phase 1 Clinical</td>
<td align="left">SM</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">Structure with Ligand</td>
<td align="left">SM</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">High-Quality Ligand</td>
<td align="left">SM</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">High-Quality Pocket</td>
<td align="left">SM</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">Med-Quality Pocket</td>
<td align="left">SM</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">Druggable Family</td>
<td align="left">SM</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">Approved Drug</td>
<td align="left">AB</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">Advanced Clinical</td>
<td align="left">AB</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">Phase 1 Clinical</td>
<td align="left">AB</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">UniProt loc high conf</td>
<td align="left">AB</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">GO CC high conf</td>
<td align="left">AB</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">UniProt loc med conf</td>
<td align="left">AB</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">UniProt SigP or TMHMM</td>
<td align="left">AB</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">GO CC med conf</td>
<td align="left">AB</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">Human Protein Atlas loc</td>
<td align="left">AB</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">Approved Drug</td>
<td align="left">PR</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">Advanced Clinical</td>
<td align="left">PR</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">Phase 1 Clinical</td>
<td align="left">PR</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">Literature</td>
<td align="left">PR</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">UniProt Ubiquitination</td>
<td align="left">PR</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">Database Ubiquitination</td>
<td align="left">PR</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">Half-life Data</td>
<td align="left">PR</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">Small Molecule Binder</td>
<td align="left">PR</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">Approved Drug</td>
<td align="left">OC</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">Advanced Clinical</td>
<td align="left">OC</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">Phase 1 Clinical</td>
<td align="left">OC</td>
<td align="left">FALSE</td>
</tr>
</tbody>
</table>
</td>
</tr></tbody></table>
</td>
</tr></tbody></table>
</td>
</tr></tbody>
</table>
</div>
</div>
<div class="section level2" number="5">
<h2 id="setup-of-gaawr2">
<span class="header-section-number">5</span> Setup of gaawr2<a class="anchor" aria-label="anchor" href="#setup-of-gaawr2"></a>
</h2>
<p>As with other R packages, it includes data examples, customized functions, documentation and featured articles.</p>
<p>The relevant scripts are with <code>inst/scripts</code> directory in the source package. Briefly,</p>
<ul>
<li>
<code>gaawr2.R</code> creates the package in R.</li>
<li>
<code>github.sh</code> creates <code>gaawr2</code> at GitHub from the command line.</li>
<li>
<code>pkgdown.sh</code> makes a pkgdown-style package and this vignette is set to be processed with the <code>bookdown</code> package.</li>
<li>
<code>docs.sh</code> adds, commits and pushes files to GitHub.</li>
</ul>
<p>Note that for creation of the GitHub repository, an SSH key is already in place. In order for <code>pkgdown.sh</code> to function well, all required files such as <code>nature-genetics.csl</code> need to be available.</p>
<p>A GitHub login is still necessary to enable web pages, so that this can be accessed as <a href="https://jinghuazhao.github.io/gaawr2/" class="uri">https://jinghuazhao.github.io/gaawr2/</a>.</p>
<p>Upon use <code>pkgdown</code>, an article can be seen as <a href="https://jinghuazhao.github.io/gaawr2/articles/index.html" class="uri">https://jinghuazhao.github.io/gaawr2/articles/index.html</a>.</p>
<p>We carry on adding files such as <code>NEWS.md</code> and <code>_pkgdown.yml</code>.</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-malone10" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">Malone, J. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btq099" class="external-link">Modeling sample variables with an experimental factor ontology</a>. <em>Bioinformatics</em> <strong>26</strong>, 1112–1118 (2010).</div>
</div>
<div id="ref-suhre20" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">Suhre, K., McCarthy, M. I. &amp; Schwenk, J. M. Genetics meets proteomics: Perspectives for large population-based studies. <em>Nat Rev Genet</em> (2020) doi:<a href="https://doi.org/10.1038/s41576-020-0268-2" class="external-link">10.1038/s41576-020-0268-2</a>.</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Crafted by <a href="https://jinghuazhao.github.io" class="external-link">Jing Hua Zhao</a>. Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Provided without <strong>any warranty</strong>.</p>
</div>

    </footer>
</div>





  </body>
</html>
