<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Genetic Association Analysis with R (II) • gaawr2</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Genetic Association Analysis with R (II)">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gaawr2</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fa-home"></span></a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><h6 class="dropdown-header" data-toc-skip>Vignettes:</h6></li>
    <li><a class="dropdown-item" href="../articles/gaawr2.html">Genetic Association Analysis with R (II)</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Articles:</h6></li>
    <li><a class="dropdown-item" href="../articles/web.html">Web facilities</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/gaawr2.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/jinghuazhao/gaawr2"><span class="fa fab fa-github"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://jinghuazhao.github.io/R/"><span class="fa fab fa-r-project"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">


<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Genetic Association Analysis with R (II)</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/jinghuazhao/gaawr2/blob/HEAD/vignettes/gaawr2.Rmd" class="external-link"><code>vignettes/gaawr2.Rmd</code></a></small>
      <div class="d-none name"><code>gaawr2.Rmd</code></div>
    </div>

    
    
<p>This package intends to gather information, meta-data and relevant scripts in genetic association analysis as a companion to a Henry-Stewart talk.</p>
<p>The package is adapted from earlier package development work, particularly pQTLdata.</p>
<div class="section level2" number="1">
<h2 id="hello-welcome-to-gaawr2">
<span class="header-section-number">1</span> Hello, welcome to gaawr2!<a class="anchor" aria-label="anchor" href="#hello-welcome-to-gaawr2"></a>
</h2>
<p>We start this session with several ways of printting a welcome message.</p>
<div class="section level3" number="1.1">
<h3 id="r">
<span class="header-section-number">1.1</span> R<a class="anchor" aria-label="anchor" href="#r"></a>
</h3>
<p>The usual and the simplest way is to execte code from the R command line prompt,</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="st">"Hello, welcome to gaawr2!"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Hello, welcome to gaawr2!"</span></span></code></pre></div>
<p>As we are now in a big data era, one might works from a Linux terminal,</p>
</div>
<div class="section level3" number="1.2">
<h3 id="linux-r">
<span class="header-section-number">1.2</span> Linux –&gt; R<a class="anchor" aria-label="anchor" href="#linux-r"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="bu">export</span> <span class="va">message</span><span class="op">=</span><span class="st">"Hello, welcome to gaawr2!"</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="ex">Rscript</span> <span class="at">-e</span> <span class="st">'message=Sys.getenv("message");print(message)'</span></span></code></pre></div>
<p>so the message is wrapped in an environment variable called <code>message</code> and grabbed into R.</p>
</div>
<div class="section level3" number="1.3">
<h3 id="linux-r-linux">
<span class="header-section-number">1.3</span> Linux –&gt; R –&gt; Linux<a class="anchor" aria-label="anchor" href="#linux-r-linux"></a>
</h3>
<p>This is somewhat unusal but I found extremely useful, notably in its use of R as part of a chain of commands, connected through the pipe
operator (<code>|</code>) and in our case there are more than one of them therefore split into many lines while informing the Linux they are in the
same chain, i.e., (<code>\</code>),</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> Hello, welcome to gaawr2 <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="ex">Rscript</span> <span class="at">-e</span> <span class="st">'</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="st">   message &lt;- scan("stdin", what = "", sep = "\n", quiet = TRUE);</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="st">   write.table(message, col.names = FALSE, row.names = FALSE, quote = FALSE)</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="st">'</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span></span></code></pre></div>
<p>Later, we will use some more useful examples on this.</p>
<p>Note that R provides its own batch command to run an R script, i.e., <code>R CMD BATCH hello.R</code> assuming our message is contained <code>hello.R</code>.</p>
</div>
</div>
<div class="section level2" number="2">
<h2 id="operators-and-packages">
<span class="header-section-number">2</span> Operators and packages<a class="anchor" aria-label="anchor" href="#operators-and-packages"></a>
</h2>
<p>The operators are intuitive in many ways, but we would like to highlight two types of them,</p>
<ul>
<li>the scope operator (<code>::</code>) is useful since user executes command from a particular package without loading it, which is usually faster.</li>
<li>the native (<code>|&gt;</code>) and contributed (<code>%&gt;%</code>) pipe operators which enable a chained of operations, the latter popularized from R <strong>magrittr</strong> and <strong>dplyr</strong> packages</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sumstats</span> <span class="op">&lt;-</span> <span class="va">mtcars</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">mpg</span> <span class="op">&gt;</span> <span class="fl">20</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">cyl</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span>avg_hp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">hp</span><span class="op">)</span>, avg_wt <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">wt</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">sumstats</span>,caption<span class="op">=</span><span class="st">"Mean horse power and weight by Cylinder for mpg over 20"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>
<span id="tab:ops">Table 2.1: </span>Mean horse power and weight by Cylinder for mpg over 20</caption>
<thead><tr class="header">
<th align="right">cyl</th>
<th align="right">avg_hp</th>
<th align="right">avg_wt</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">4</td>
<td align="right">82.63636</td>
<td align="right">2.285727</td>
</tr>
<tr class="even">
<td align="right">6</td>
<td align="right">110.00000</td>
<td align="right">2.903333</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">sumstats</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">cyl</span><span class="op">)</span>, y <span class="op">=</span> <span class="va">avg_hp</span>, fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">cyl</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>    title <span class="op">=</span> <span class="st">"Average Horsepower by Cylinder Count (mpg &gt; 20)"</span>,</span>
<span>    x <span class="op">=</span> <span class="st">"Number of Cylinders"</span>,</span>
<span>    y <span class="op">=</span> <span class="st">"Average Horsepower"</span></span>
<span>  <span class="op">)</span> <span class="op">+</span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html" class="external-link">scale_fill_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Set2"</span>, name <span class="op">=</span> <span class="st">"Cylinders"</span><span class="op">)</span></span></code></pre></div>
<p><img src="gaawr2%2Fops-1.png" width="672" style="display:block; margin: auto"></p>
<p>Two most often used approaches to make R packages available is through</p>
<ul>
<li>
<code><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages()</a></code> which is a standard way to install from the Comprehensive R Archive Network (CRAN) hosting many user-contributed packages.</li>
<li><code>BiocManager::install() which is an updated command to install package from the Bioconductor project.</code></li>
</ul>
</div>
<div class="section level2" number="3">
<h2 id="data-analysis">
<span class="header-section-number">3</span> Data analysis<a class="anchor" aria-label="anchor" href="#data-analysis"></a>
</h2>
<p>Topics vary from those heavily featured in candidate gene studies, such as Hardy-Weinberg equilirium (HWE), to their counterparts in
Genomewide association study (GWAS) such as various types of association statistics, QQ/Manhattan/local association plots.</p>
<p>There has been a lot of interests in machine learning (ML), artificial language (AI), including deep learning (DL), just to add one more
acronym, the bulk of which is also readily available.</p>
<div class="section level3" number="3.1">
<h3 id="hardyweinberg">
<span class="header-section-number">3.1</span> HardyWeinberg<a class="anchor" aria-label="anchor" href="#hardyweinberg"></a>
</h3>
<p>We set to run through the package for HWE. Three data sources are used: MN blood group in the documentation,
a chromosome X SNP and a HLA/DQR,</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>width<span class="op">=</span><span class="fl">200</span><span class="op">)</span></span>
<span><span class="co"># MN blood group</span></span>
<span><span class="va">SNP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>MM <span class="op">=</span> <span class="fl">298</span>, MN <span class="op">=</span> <span class="fl">489</span>, NN <span class="op">=</span> <span class="fl">213</span><span class="op">)</span></span>
<span><span class="fu">HardyWeinberg</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HardyWeinberg/man/maf.html" class="external-link">maf</a></span><span class="op">(</span><span class="va">SNP</span><span class="op">)</span></span>
<span><span class="co">#&gt;      N </span></span>
<span><span class="co">#&gt; 0.4575</span></span>
<span><span class="fu">HardyWeinberg</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HardyWeinberg/man/HWTernaryPlot.html" class="external-link">HWTernaryPlot</a></span><span class="op">(</span><span class="va">SNP</span>,region<span class="op">=</span><span class="fl">0</span>,hwcurve<span class="op">=</span><span class="cn">FALSE</span>,grid<span class="op">=</span><span class="cn">TRUE</span>,markercol<span class="op">=</span><span class="st">"blue"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:hwe"></span>
<img src="gaawr2%2Fhwe-1.png" alt="SNP ternary plot" width="1497.6" style="display:block; margin: auto"><p class="caption">
Figure 3.1: SNP ternary plot
</p>
</div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">HardyWeinberg</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HardyWeinberg/man/HWChisq.html" class="external-link">HWChisq</a></span><span class="op">(</span><span class="va">SNP</span>, cc <span class="op">=</span> <span class="fl">0</span>, verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Chi-square test for Hardy-Weinberg equilibrium (autosomal)</span></span>
<span><span class="co">#&gt; Chi2 =  0.2214896 DF =  1 p-value =  0.6379073 D =  -3.69375 f =  0.01488253</span></span>
<span><span class="co"># Chromosome X</span></span>
<span><span class="va">xSNP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>A<span class="op">=</span><span class="fl">10</span>, B<span class="op">=</span><span class="fl">20</span>, AA<span class="op">=</span><span class="fl">30</span>, AB<span class="op">=</span><span class="fl">20</span>, BB<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu">HardyWeinberg</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HardyWeinberg/man/HWChisq.html" class="external-link">HWChisq</a></span><span class="op">(</span><span class="va">xSNP</span>,cc<span class="op">=</span><span class="fl">0</span>,x.linked<span class="op">=</span><span class="cn">TRUE</span>,verbose<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Chi-square test for Hardy-Weinberg equilibrium (X-chromosomal)</span></span>
<span><span class="co">#&gt; Chi2 =  14.86111 DF = 2 p-value =  0.0005928581 D =  NA f =  0.25</span></span>
<span><span class="co"># HLA/DQR</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jinghuazhao/R" class="external-link">gap</a></span><span class="op">)</span></span>
<span><span class="va">DQR</span> <span class="op">&lt;-</span> <span class="va">hla</span><span class="op">[</span>,<span class="fl">3</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span>
<span><span class="va">a1</span> <span class="op">&lt;-</span> <span class="va">DQR</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">a2</span> <span class="op">&lt;-</span> <span class="va">DQR</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="va">GenotypeCounts</span> <span class="op">&lt;-</span> <span class="fu">HardyWeinberg</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HardyWeinberg/man/AllelesToTriangular.html" class="external-link">AllelesToTriangular</a></span><span class="op">(</span><span class="va">a1</span>,<span class="va">a2</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">GenotypeCounts</span>,caption<span class="op">=</span><span class="st">"Genotype distribution of DQR"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>
<span id="tab:hwe">Table 3.1: </span>Genotype distribution of DQR</caption>
<colgroup>
<col width="4%">
<col width="3%">
<col width="3%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="3%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">A0</th>
<th align="right">A1</th>
<th align="right">A10</th>
<th align="right">A11</th>
<th align="right">A12</th>
<th align="right">A13</th>
<th align="right">A14</th>
<th align="right">A15</th>
<th align="right">A16</th>
<th align="right">A17</th>
<th align="right">A18</th>
<th align="right">A19</th>
<th align="right">A2</th>
<th align="right">A20</th>
<th align="right">A21</th>
<th align="right">A22</th>
<th align="right">A23</th>
<th align="right">A24</th>
<th align="right">A25</th>
<th align="right">A3</th>
<th align="right">A4</th>
<th align="right">A6</th>
<th align="right">A7</th>
<th align="right">A8</th>
<th align="right">A9</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">A0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A1</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A10</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A11</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A12</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A13</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A14</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A15</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A16</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A17</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A18</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A19</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A20</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A21</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A22</td>
<td align="right">0</td>
<td align="right">13</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A23</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A24</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A25</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A3</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A4</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="right">8</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">4</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A6</td>
<td align="right">0</td>
<td align="right">6</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">7</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">11</td>
<td align="right">7</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A7</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">A8</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">9</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A9</td>
<td align="right">0</td>
<td align="right">8</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">2</td>
<td align="right">3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">5</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">3</td>
<td align="right">15</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">7</td>
<td align="right">10</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="right">2</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">HardyWeinberg</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HardyWeinberg/man/HWPerm.mult.html" class="external-link">HWPerm.mult</a></span><span class="op">(</span><span class="va">GenotypeCounts</span>,nperm<span class="op">=</span><span class="fl">1000</span><span class="op">)</span></span>
<span><span class="co">#&gt; Permutation test for Hardy-Weinberg equilibrium (autosomal).</span></span>
<span><span class="co">#&gt; 25 alleles detected.</span></span>
<span><span class="co">#&gt; Observed statistic: 1.461325e-84   1000 permutations. p-value: 0.023</span></span>
<span><span class="fu">HardyWeinberg</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/HardyWeinberg/man/HWStr.html" class="external-link">HWStr</a></span><span class="op">(</span><span class="va">hla</span><span class="op">[</span>,<span class="fl">3</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span>,test<span class="op">=</span><span class="st">"permutation"</span>,nperm<span class="op">=</span><span class="fl">1000</span><span class="op">)</span></span>
<span><span class="co">#&gt; 1 STRs detected.</span></span>
<span><span class="co">#&gt;      STR   N Nt     MinorAF   MajorAF        Ho        He       Hp  pval</span></span>
<span><span class="co">#&gt; 1 DQR.a1 271 25 0.001845018 0.1494465 0.8856089 0.9097643 2.654949 0.026</span></span></code></pre></div>
</div>
<div class="section level3" number="3.2">
<h3 id="snpassoc">
<span class="header-section-number">3.2</span> SNPassoc<a class="anchor" aria-label="anchor" href="#snpassoc"></a>
</h3>
<p>The package implements procedures which are appropriate for candidate gene association analysis, under a variety of genetic models.</p>
<p>Fist we look at some meta-data, include HWE.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>width<span class="op">=</span><span class="fl">200</span>,digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">asthma</span>, package <span class="op">=</span> <span class="st">"SNPassoc"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">asthma</span>, list.len<span class="op">=</span><span class="fl">8</span><span class="op">)</span></span>
<span><span class="co">#&gt; 'data.frame':    1578 obs. of  57 variables:</span></span>
<span><span class="co">#&gt;  $ country    : Factor w/ 10 levels "Australia","Belgium",..: 5 5 5 5 5 5 5 5 5 5 ...</span></span>
<span><span class="co">#&gt;  $ gender     : Factor w/ 2 levels "Females","Males": 2 2 2 1 1 1 1 2 1 1 ...</span></span>
<span><span class="co">#&gt;  $ age        : num  42.8 50.2 46.7 47.9 48.4 ...</span></span>
<span><span class="co">#&gt;  $ bmi        : num  20.1 24.7 27.7 33.3 25.2 ...</span></span>
<span><span class="co">#&gt;  $ smoke      : int  1 0 0 0 0 1 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  $ casecontrol: int  0 0 0 0 1 0 0 0 0 0 ...</span></span>
<span><span class="co">#&gt;  $ rs4490198  : Factor w/ 3 levels "AA","AG","GG": 3 3 3 2 2 2 3 2 2 2 ...</span></span>
<span><span class="co">#&gt;  $ rs4849332  : Factor w/ 3 levels "GG","GT","TT": 3 2 3 2 1 2 3 3 2 1 ...</span></span>
<span><span class="co">#&gt;   [list output truncated]</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">asthma</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span>,caption<span class="op">=</span><span class="st">"First three records &amp; two SNPs"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>
<span id="tab:snpassoc1">Table 3.2: </span>First three records &amp; two SNPs</caption>
<thead><tr class="header">
<th align="left">country</th>
<th align="left">gender</th>
<th align="right">age</th>
<th align="right">bmi</th>
<th align="right">smoke</th>
<th align="right">casecontrol</th>
<th align="left">rs4490198</th>
<th align="left">rs4849332</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Germany</td>
<td align="left">Males</td>
<td align="right">43</td>
<td align="right">20</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="left">GG</td>
<td align="left">TT</td>
</tr>
<tr class="even">
<td align="left">Germany</td>
<td align="left">Males</td>
<td align="right">50</td>
<td align="right">25</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">GG</td>
<td align="left">GT</td>
</tr>
<tr class="odd">
<td align="left">Germany</td>
<td align="left">Males</td>
<td align="right">47</td>
<td align="right">28</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">GG</td>
<td align="left">TT</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">snpCols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">asthma</span><span class="op">)</span><span class="op">[</span><span class="fl">6</span><span class="op">+</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">snps</span> <span class="op">&lt;-</span> <span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/setupSNP.html" class="external-link">setupSNP</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">asthma</span><span class="op">[</span><span class="va">snpCols</span><span class="op">]</span>, colSNPs<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">snpCols</span><span class="op">)</span>, sep<span class="op">=</span><span class="st">""</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">snps</span>, print<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt;           alleles major.allele.freq HWE  missing (%)</span></span>
<span><span class="co">#&gt; rs4490198 A/G     59                0.17 0.6        </span></span>
<span><span class="co">#&gt; rs4849332 G/T     62                0.52 0.1</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">snps</span>, <span class="va">head</span><span class="op">)</span></span>
<span><span class="co">#&gt; $rs4490198</span></span>
<span><span class="co">#&gt; [1] G/G G/G G/G A/G A/G A/G</span></span>
<span><span class="co">#&gt; Genotypes: A/A A/G G/G</span></span>
<span><span class="co">#&gt; Alleles:  G A </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $rs4849332</span></span>
<span><span class="co">#&gt; [1] T/T G/T T/T G/T G/G G/T</span></span>
<span><span class="co">#&gt; Genotypes: G/G G/T T/T</span></span>
<span><span class="co">#&gt; Alleles:  T G</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">snps</span>, <span class="va">summary</span><span class="op">)</span></span>
<span><span class="co">#&gt; $rs4490198</span></span>
<span><span class="co">#&gt; Genotypes: </span></span>
<span><span class="co">#&gt;      frequency percentage</span></span>
<span><span class="co">#&gt; A/A        562         36</span></span>
<span><span class="co">#&gt; A/G        731         47</span></span>
<span><span class="co">#&gt; G/G        275         18</span></span>
<span><span class="co">#&gt; NA's        10           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Alleles: </span></span>
<span><span class="co">#&gt;      frequency percentage</span></span>
<span><span class="co">#&gt; G         1281         41</span></span>
<span><span class="co">#&gt; A         1855         59</span></span>
<span><span class="co">#&gt; NA's        20           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; HWE (p value): 0.17 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $rs4849332</span></span>
<span><span class="co">#&gt; Genotypes: </span></span>
<span><span class="co">#&gt;      frequency percentage</span></span>
<span><span class="co">#&gt; G/G        609         39</span></span>
<span><span class="co">#&gt; G/T        732         46</span></span>
<span><span class="co">#&gt; T/T        236         15</span></span>
<span><span class="co">#&gt; NA's         1           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Alleles: </span></span>
<span><span class="co">#&gt;      frequency percentage</span></span>
<span><span class="co">#&gt; T         1204         38</span></span>
<span><span class="co">#&gt; G         1950         62</span></span>
<span><span class="co">#&gt; NA's         2           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; HWE (p value): 0.52</span></span>
<span><span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/tableHWE.html" class="external-link">tableHWE</a></span><span class="op">(</span><span class="va">snps</span><span class="op">)</span></span>
<span><span class="co">#&gt;           HWE (p value) flag</span></span>
<span><span class="co">#&gt; rs4490198 0.17              </span></span>
<span><span class="co">#&gt; rs4849332 0.52</span></span></code></pre></div>
<p>where variable <code>snpCols</code> skips six columns non-SNP data for two SNPs.</p>
<p>We then turn to genetic models for the first one,</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">asthma.snps</span> <span class="op">&lt;-</span> <span class="va">asthma</span> <span class="op">%&gt;%</span></span>
<span>               <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>cc<span class="op">=</span><span class="va">casecontrol</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>               <span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/setupSNP.html" class="external-link">setupSNP</a></span><span class="op">(</span>colSNPs<span class="op">=</span><span class="fl">7</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span>, sep<span class="op">=</span><span class="st">""</span><span class="op">)</span></span>
<span><span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/association.html" class="external-link">association</a></span><span class="op">(</span><span class="va">bmi</span> <span class="op">~</span> <span class="va">rs4490198</span>, <span class="va">asthma.snps</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; SNP: rs4490198  adjusted by: </span></span>
<span><span class="co">#&gt;                 n   me    se     dif  lower upper p-value  AIC</span></span>
<span><span class="co">#&gt; Codominant                                                    </span></span>
<span><span class="co">#&gt; A/A           558 25.5 0.176  0.0000                0.885 9015</span></span>
<span><span class="co">#&gt; A/G           725 25.6 0.174  0.0217 -0.461 0.505             </span></span>
<span><span class="co">#&gt; G/G           273 25.4 0.237 -0.1299 -0.763 0.504             </span></span>
<span><span class="co">#&gt; Dominant                                                      </span></span>
<span><span class="co">#&gt; A/A           558 25.5 0.176  0.0000                0.932 9013</span></span>
<span><span class="co">#&gt; A/G-G/G       998 25.5 0.142 -0.0198 -0.473 0.434             </span></span>
<span><span class="co">#&gt; Recessive                                                     </span></span>
<span><span class="co">#&gt; A/A-A/G      1283 25.6 0.125  0.0000                0.626 9013</span></span>
<span><span class="co">#&gt; G/G           273 25.4 0.237 -0.1421 -0.714 0.429             </span></span>
<span><span class="co">#&gt; Overdominant                                                  </span></span>
<span><span class="co">#&gt; A/A-G/G       831 25.5 0.142  0.0000                0.772 9013</span></span>
<span><span class="co">#&gt; A/G           725 25.6 0.174  0.0643 -0.371 0.500             </span></span>
<span><span class="co">#&gt; log-Additive                                                  </span></span>
<span><span class="co">#&gt; 0,1,2                        -0.0502 -0.357 0.257   0.749 9013</span></span>
<span><span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/association.html" class="external-link">association</a></span><span class="op">(</span><span class="va">cc</span> <span class="op">~</span> <span class="va">rs4490198</span>, data<span class="op">=</span><span class="va">asthma.snps</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; SNP: rs4490198  adjusted by: </span></span>
<span><span class="co">#&gt;                 0    %   1    %   OR lower upper p-value  AIC</span></span>
<span><span class="co">#&gt; Codominant                                                   </span></span>
<span><span class="co">#&gt; A/A           449 36.5 113 33.4 1.00               0.528 1639</span></span>
<span><span class="co">#&gt; A/G           565 45.9 166 49.1 1.17  0.89  1.53             </span></span>
<span><span class="co">#&gt; G/G           216 17.6  59 17.5 1.09  0.76  1.55             </span></span>
<span><span class="co">#&gt; Dominant                                                     </span></span>
<span><span class="co">#&gt; A/A           449 36.5 113 33.4 1.00               0.295 1638</span></span>
<span><span class="co">#&gt; A/G-G/G       781 63.5 225 66.6 1.14  0.89  1.48             </span></span>
<span><span class="co">#&gt; Recessive                                                    </span></span>
<span><span class="co">#&gt; A/A-A/G      1014 82.4 279 82.5 1.00               0.964 1639</span></span>
<span><span class="co">#&gt; G/G           216 17.6  59 17.5 0.99  0.72  1.36             </span></span>
<span><span class="co">#&gt; Overdominant                                                 </span></span>
<span><span class="co">#&gt; A/A-G/G       665 54.1 172 50.9 1.00               0.300 1638</span></span>
<span><span class="co">#&gt; A/G           565 45.9 166 49.1 1.14  0.89  1.45             </span></span>
<span><span class="co">#&gt; log-Additive                                                 </span></span>
<span><span class="co">#&gt; 0,1,2        1230 78.4 338 21.6 1.06  0.90  1.26   0.495 1638</span></span>
<span><span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/association.html" class="external-link">association</a></span><span class="op">(</span><span class="va">cc</span> <span class="op">~</span> <span class="va">rs4490198</span> <span class="op">+</span> <span class="va">country</span> <span class="op">+</span> <span class="va">smoke</span>, <span class="va">asthma.snps</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; SNP: rs4490198  adjusted by: country smoke </span></span>
<span><span class="co">#&gt;                 0    %   1    %   OR lower upper p-value  AIC</span></span>
<span><span class="co">#&gt; Codominant                                                   </span></span>
<span><span class="co">#&gt; A/A           448 36.6 112 33.2 1.00               0.541 1403</span></span>
<span><span class="co">#&gt; A/G           563 46.0 166 49.3 1.18  0.88  1.59             </span></span>
<span><span class="co">#&gt; G/G           213 17.4  59 17.5 1.08  0.73  1.60             </span></span>
<span><span class="co">#&gt; Dominant                                                     </span></span>
<span><span class="co">#&gt; A/A           448 36.6 112 33.2 1.00               0.317 1401</span></span>
<span><span class="co">#&gt; A/G-G/G       776 63.4 225 66.8 1.15  0.87  1.52             </span></span>
<span><span class="co">#&gt; Recessive                                                    </span></span>
<span><span class="co">#&gt; A/A-A/G      1011 82.6 278 82.5 1.00               0.909 1402</span></span>
<span><span class="co">#&gt; G/G           213 17.4  59 17.5 0.98  0.69  1.39             </span></span>
<span><span class="co">#&gt; Overdominant                                                 </span></span>
<span><span class="co">#&gt; A/A-G/G       661 54.0 171 50.7 1.00               0.298 1401</span></span>
<span><span class="co">#&gt; A/G           563 46.0 166 49.3 1.15  0.88  1.50             </span></span>
<span><span class="co">#&gt; log-Additive                                                 </span></span>
<span><span class="co">#&gt; 0,1,2        1224 78.4 337 21.6 1.06  0.88  1.28   0.538 1402</span></span>
<span><span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/association.html" class="external-link">association</a></span><span class="op">(</span><span class="va">cc</span> <span class="op">~</span> <span class="va">rs4490198</span> <span class="op">+</span> <span class="fu">survival</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/survival/man/strata.html" class="external-link">strata</a></span><span class="op">(</span><span class="va">gender</span><span class="op">)</span>, <span class="va">asthma.snps</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; SNP: rs4490198  adjusted by: survival::strata(gender) </span></span>
<span><span class="co">#&gt;                 0    %   1    %   OR lower upper p-value  AIC</span></span>
<span><span class="co">#&gt; Codominant                                                   </span></span>
<span><span class="co">#&gt; A/A           449 36.5 113 33.4 1.00               0.552 1630</span></span>
<span><span class="co">#&gt; A/G           565 45.9 166 49.1 1.16  0.89  1.52             </span></span>
<span><span class="co">#&gt; G/G           216 17.6  59 17.5 1.10  0.77  1.57             </span></span>
<span><span class="co">#&gt; Dominant                                                     </span></span>
<span><span class="co">#&gt; A/A           449 36.5 113 33.4 1.00               0.295 1629</span></span>
<span><span class="co">#&gt; A/G-G/G       781 63.5 225 66.6 1.15  0.89  1.48             </span></span>
<span><span class="co">#&gt; Recessive                                                    </span></span>
<span><span class="co">#&gt; A/A-A/G      1014 82.4 279 82.5 1.00               0.942 1630</span></span>
<span><span class="co">#&gt; G/G           216 17.6  59 17.5 1.01  0.74  1.39             </span></span>
<span><span class="co">#&gt; Overdominant                                                 </span></span>
<span><span class="co">#&gt; A/A-G/G       665 54.1 172 50.9 1.00               0.344 1629</span></span>
<span><span class="co">#&gt; A/G           565 45.9 166 49.1 1.12  0.88  1.43             </span></span>
<span><span class="co">#&gt; log-Additive                                                 </span></span>
<span><span class="co">#&gt; 0,1,2        1230 78.4 338 21.6 1.07  0.90  1.27   0.455 1629</span></span>
<span><span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/association.html" class="external-link">association</a></span><span class="op">(</span><span class="va">cc</span> <span class="op">~</span> <span class="va">rs4490198</span>, <span class="va">asthma.snps</span>, subset<span class="op">=</span><span class="va">country</span><span class="op">==</span><span class="st">"Spain"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; SNP: rs4490198  adjusted by: </span></span>
<span><span class="co">#&gt;                0    %  1    %   OR lower upper p-value AIC</span></span>
<span><span class="co">#&gt; Codominant                                                </span></span>
<span><span class="co">#&gt; A/A          141 43.3 18 37.5 1.00               0.626 292</span></span>
<span><span class="co">#&gt; A/G          145 44.5 22 45.8 1.19  0.61  2.31            </span></span>
<span><span class="co">#&gt; G/G           40 12.3  8 16.7 1.57  0.63  3.87            </span></span>
<span><span class="co">#&gt; Dominant                                                  </span></span>
<span><span class="co">#&gt; A/A          141 43.3 18 37.5 1.00               0.449 290</span></span>
<span><span class="co">#&gt; A/G-G/G      185 56.7 30 62.5 1.27  0.68  2.37            </span></span>
<span><span class="co">#&gt; Recessive                                                 </span></span>
<span><span class="co">#&gt; A/A-A/G      286 87.7 40 83.3 1.00               0.410 290</span></span>
<span><span class="co">#&gt; G/G           40 12.3  8 16.7 1.43  0.62  3.27            </span></span>
<span><span class="co">#&gt; Overdominant                                              </span></span>
<span><span class="co">#&gt; A/A-G/G      181 55.5 26 54.2 1.00               0.860 291</span></span>
<span><span class="co">#&gt; A/G          145 44.5 22 45.8 1.06  0.57  1.94            </span></span>
<span><span class="co">#&gt; log-Additive                                              </span></span>
<span><span class="co">#&gt; 0,1,2        326 87.2 48 12.8 1.24  0.80  1.91   0.339 290</span></span>
<span><span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/association.html" class="external-link">association</a></span><span class="op">(</span><span class="va">cc</span> <span class="op">~</span> <span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/inheritance.html" class="external-link">dominant</a></span><span class="op">(</span><span class="va">rs4490198</span><span class="op">)</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">smoke</span><span class="op">)</span>, data<span class="op">=</span><span class="va">asthma.snps</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;       SNP: SNPassoc::dominant(rs4490198  adjusted by: </span></span>
<span><span class="co">#&gt;  Interaction </span></span>
<span><span class="co">#&gt; ---------------------</span></span>
<span><span class="co">#&gt;           0       OR lower upper   1      OR lower upper</span></span>
<span><span class="co">#&gt; A/A     293  84 1.00    NA    NA 155 28 0.63  0.39  1.01</span></span>
<span><span class="co">#&gt; A/G-G/G 539 172 1.11  0.83   1.5 237 53 0.78  0.53  1.15</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; p interaction: 0.7 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  factor(smoke) within SNPassoc::dominant(rs4490198 </span></span>
<span><span class="co">#&gt; ---------------------</span></span>
<span><span class="co">#&gt; A/A </span></span>
<span><span class="co">#&gt;     0  1   OR lower upper</span></span>
<span><span class="co">#&gt; 0 293 84 1.00    NA    NA</span></span>
<span><span class="co">#&gt; 1 155 28 0.63  0.39  1.01</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A/G-G/G </span></span>
<span><span class="co">#&gt;     0   1  OR lower upper</span></span>
<span><span class="co">#&gt; 0 539 172 1.0    NA    NA</span></span>
<span><span class="co">#&gt; 1 237  53 0.7   0.5  0.99</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; p trend: 0.7 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  SNPassoc::dominant(rs4490198 within factor(smoke) </span></span>
<span><span class="co">#&gt; ---------------------</span></span>
<span><span class="co">#&gt; 0 </span></span>
<span><span class="co">#&gt;           0   1   OR lower upper</span></span>
<span><span class="co">#&gt; A/A     293  84 1.00    NA    NA</span></span>
<span><span class="co">#&gt; A/G-G/G 539 172 1.11  0.83   1.5</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 1 </span></span>
<span><span class="co">#&gt;           0  1   OR lower upper</span></span>
<span><span class="co">#&gt; A/A     155 28 1.00    NA    NA</span></span>
<span><span class="co">#&gt; A/G-G/G 237 53 1.24  0.75  2.04</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; p trend: 0.7</span></span>
<span><span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/association.html" class="external-link">association</a></span><span class="op">(</span><span class="va">cc</span> <span class="op">~</span> <span class="va">rs4490198</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">rs11123242</span><span class="op">)</span>, data<span class="op">=</span><span class="va">asthma.snps</span>, model.interaction <span class="op">=</span> <span class="st">"dominant"</span> <span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;       SNP: rs4490198  adjusted by: </span></span>
<span><span class="co">#&gt;  Interaction </span></span>
<span><span class="co">#&gt; ---------------------</span></span>
<span><span class="co">#&gt;         C/C       OR lower upper C/T       OR lower upper  0  1  T/T lower upper</span></span>
<span><span class="co">#&gt; A/A     448 113 1.00    NA    NA   1   0 0.00  0.00    NA  0  0   NA    NA    NA</span></span>
<span><span class="co">#&gt; A/G-G/G 371 109 1.16  0.87  1.57 365 102 1.11  0.82   1.5 39 14 1.42  0.75  2.71</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; p interaction: 0.5 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  factor(rs11123242) within rs4490198 </span></span>
<span><span class="co">#&gt; ---------------------</span></span>
<span><span class="co">#&gt; A/A </span></span>
<span><span class="co">#&gt;       0   1 OR lower upper</span></span>
<span><span class="co">#&gt; C/C 448 113  1    NA    NA</span></span>
<span><span class="co">#&gt; C/T   1   0  0     0    NA</span></span>
<span><span class="co">#&gt; T/T   0   0 NA    NA    NA</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A/G-G/G </span></span>
<span><span class="co">#&gt;       0   1   OR lower upper</span></span>
<span><span class="co">#&gt; C/C 371 109 1.00    NA    NA</span></span>
<span><span class="co">#&gt; C/T 365 102 0.95  0.70  1.29</span></span>
<span><span class="co">#&gt; T/T  39  14 1.22  0.64  2.33</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; p trend: 0.5 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  rs4490198 within factor(rs11123242) </span></span>
<span><span class="co">#&gt; ---------------------</span></span>
<span><span class="co">#&gt; C/C </span></span>
<span><span class="co">#&gt;           0   1   OR lower upper</span></span>
<span><span class="co">#&gt; A/A     448 113 1.00    NA    NA</span></span>
<span><span class="co">#&gt; A/G-G/G 371 109 1.16  0.87  1.57</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; C/T </span></span>
<span><span class="co">#&gt;           0   1 OR lower upper</span></span>
<span><span class="co">#&gt; A/A       1   0  1    NA    NA</span></span>
<span><span class="co">#&gt; A/G-G/G 365 102 NA     0    NA</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; T/T </span></span>
<span><span class="co">#&gt;          0  1 OR lower upper</span></span>
<span><span class="co">#&gt; A/A      0  0  1    NA    NA</span></span>
<span><span class="co">#&gt; A/G-G/G 39 14 NA    NA    NA</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; p trend: 0.5</span></span></code></pre></div>
</div>
<div class="section level3" number="3.3">
<h3 id="haplo-stats">
<span class="header-section-number">3.3</span> haplo.stats<a class="anchor" aria-label="anchor" href="#haplo-stats"></a>
</h3>
<p>This package considers haplotype estimation using EM-algorithms and genetic association under a generalized linear model (GLM).</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">asthma</span>, package <span class="op">=</span> <span class="st">"SNPassoc"</span><span class="op">)</span></span>
<span><span class="va">asthma.snps</span> <span class="op">&lt;-</span> <span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/setupSNP.html" class="external-link">setupSNP</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">asthma</span>, colSNPs<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"^rs"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">asthma</span><span class="op">)</span><span class="op">)</span>, sep<span class="op">=</span><span class="st">""</span><span class="op">)</span></span>
<span><span class="va">asthma.snps</span> <span class="op">&lt;-</span> <span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/setupSNP.html" class="external-link">setupSNP</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">asthma</span>, colSNPs<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"^rs"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">asthma</span><span class="op">)</span><span class="op">)</span>, sep<span class="op">=</span><span class="st">""</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>               <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>cc<span class="op">=</span><span class="va">casecontrol</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://analytictools.mayo.edu/research/haplo-stats/" class="external-link">haplo.stats</a></span><span class="op">)</span></span>
<span><span class="va">snpsH</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"rs714588"</span>, <span class="st">"rs1023555"</span>,  <span class="st">"rs898070"</span><span class="op">)</span></span>
<span><span class="va">genoH</span> <span class="op">&lt;-</span> <span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/makegeno.html" class="external-link">make.geno</a></span><span class="op">(</span><span class="va">asthma.snps</span>, <span class="va">snpsH</span><span class="op">)</span></span>
<span><span class="va">em</span> <span class="op">&lt;-</span> <span class="fu">haplo.stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/haplo.stats/man/haplo.em.html" class="external-link">haplo.em</a></span><span class="op">(</span><span class="va">genoH</span>, locus.label <span class="op">=</span> <span class="va">snpsH</span>, miss.val <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="cn">NA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">em</span></span>
<span><span class="co">#&gt; ======================================================================================================================================================================================================== </span></span>
<span><span class="co">#&gt;                                                                                                Haplotypes                                                                                                </span></span>
<span><span class="co">#&gt; ======================================================================================================================================================================================================== </span></span>
<span><span class="co">#&gt;   rs714588 rs1023555 rs898070 hap.freq</span></span>
<span><span class="co">#&gt; 1        1         1        1    0.041</span></span>
<span><span class="co">#&gt; 2        1         1        2    0.024</span></span>
<span><span class="co">#&gt; 3        1         2        1    0.043</span></span>
<span><span class="co">#&gt; 4        1         2        2    0.440</span></span>
<span><span class="co">#&gt; 5        2         1        1    0.083</span></span>
<span><span class="co">#&gt; 6        2         1        2    0.084</span></span>
<span><span class="co">#&gt; 7        2         2        1    0.208</span></span>
<span><span class="co">#&gt; 8        2         2        2    0.077</span></span>
<span><span class="co">#&gt; ======================================================================================================================================================================================================== </span></span>
<span><span class="co">#&gt;                                                                                                 Details                                                                                                  </span></span>
<span><span class="co">#&gt; ======================================================================================================================================================================================================== </span></span>
<span><span class="co">#&gt; lnlike =  -4103 </span></span>
<span><span class="co">#&gt; lr stat for no LD =  775 , df =  4 , p-val =  0</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">em</span><span class="op">$</span><span class="va">haplotype</span>,<span class="va">em</span><span class="op">$</span><span class="va">hap.prob</span><span class="op">)</span>,caption<span class="op">=</span><span class="st">"Haplotypes of rs714588, rs1023555, rs898070"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>(#tab:haplo.stats)Haplotypes of rs714588, rs1023555, rs898070</caption>
<thead><tr class="header">
<th align="left">rs714588</th>
<th align="left">rs1023555</th>
<th align="left">rs898070</th>
<th align="right">em$hap.prob</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="left">1</td>
<td align="left">1</td>
<td align="right">0.04</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">1</td>
<td align="left">2</td>
<td align="right">0.02</td>
</tr>
<tr class="odd">
<td align="left">1</td>
<td align="left">2</td>
<td align="left">1</td>
<td align="right">0.04</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">2</td>
<td align="left">2</td>
<td align="right">0.44</td>
</tr>
<tr class="odd">
<td align="left">2</td>
<td align="left">1</td>
<td align="left">1</td>
<td align="right">0.08</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="left">1</td>
<td align="left">2</td>
<td align="right">0.08</td>
</tr>
<tr class="odd">
<td align="left">2</td>
<td align="left">2</td>
<td align="left">1</td>
<td align="right">0.21</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="left">2</td>
<td align="left">2</td>
<td align="right">0.08</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu">haplo.stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/haplo.stats/man/haplo.glm.html" class="external-link">haplo.glm</a></span><span class="op">(</span><span class="va">cc</span> <span class="op">~</span> <span class="va">genoH</span>, data<span class="op">=</span><span class="va">asthma.snps</span>,</span>
<span>                              family<span class="op">=</span><span class="st">"binomial"</span>,</span>
<span>                              locus.label<span class="op">=</span><span class="va">snpsH</span>,</span>
<span>                              allele.lev<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">genoH</span><span class="op">)</span><span class="op">$</span><span class="va">unique.alleles</span>,</span>
<span>                              control <span class="op">=</span> <span class="fu">haplo.stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/haplo.stats/man/haplo.glm.control.html" class="external-link">haplo.glm.control</a></span><span class="op">(</span>haplo.freq.min<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mod</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Call:  haplo.stats::haplo.glm(formula = cc ~ genoH, family = "binomial", </span></span>
<span><span class="co">#&gt;     data = asthma.snps, locus.label = snpsH, control = haplo.stats::haplo.glm.control(haplo.freq.min = 0.05), </span></span>
<span><span class="co">#&gt;     allele.lev = attributes(genoH)$unique.alleles)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Coefficients:</span></span>
<span><span class="co">#&gt; (Intercept)      genoH.5      genoH.6      genoH.7      genoH.8   genoH.rare  </span></span>
<span><span class="co">#&gt;  -1.3741805    0.0900879   -0.0082069    0.0560710    0.0882132    0.1396095  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Haplotypes:</span></span>
<span><span class="co">#&gt;            rs714588 rs1023555 rs898070 hap.freq</span></span>
<span><span class="co">#&gt; genoH.5           G         A        A 0.082685</span></span>
<span><span class="co">#&gt; genoH.6           G         A        G 0.084073</span></span>
<span><span class="co">#&gt; genoH.7           G         T        A 0.207956</span></span>
<span><span class="co">#&gt; genoH.8           G         T        G 0.076886</span></span>
<span><span class="co">#&gt; genoH.rare        *         *        * 0.107905</span></span>
<span><span class="co">#&gt; haplo.base        A         T        G 0.440495</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Degrees of Freedom:  1577 Total (i.e. Null);  1572 Residual</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Null Deviance:  1645 </span></span>
<span><span class="co">#&gt;  Residual Deviance:  1643 </span></span>
<span><span class="co">#&gt;                AIC:  1655</span></span>
<span><span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/intervals.html" class="external-link">intervals</a></span><span class="op">(</span><span class="va">mod</span><span class="op">)</span></span>
<span><span class="co">#&gt;                freq   or   95%   C.I.   P-val </span></span>
<span><span class="co">#&gt;          ATG 0.4405   1.00 Reference haplotype </span></span>
<span><span class="co">#&gt;          GAA 0.0827   1.09 (   0.77 -   1.56 )  0.6160 </span></span>
<span><span class="co">#&gt;          GAG 0.0841   0.99 (   0.69 -   1.42 )  0.9642 </span></span>
<span><span class="co">#&gt;          GTA 0.2080   1.06 (   0.84 -   1.34 )  0.6379 </span></span>
<span><span class="co">#&gt;          GTG 0.0769   1.09 (   0.76 -   1.58 )  0.6367 </span></span>
<span><span class="co">#&gt;   genoH.rare 0.1079   1.15 (   0.83 -   1.59 )  0.3968</span></span>
<span><span class="va">snpsH2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/labels.html" class="external-link">labels</a></span><span class="op">(</span><span class="va">asthma.snps</span><span class="op">)</span><span class="op">[</span><span class="fl">6</span><span class="op">:</span><span class="fl">15</span><span class="op">]</span></span>
<span><span class="va">genoH2</span> <span class="op">&lt;-</span> <span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/makegeno.html" class="external-link">make.geno</a></span><span class="op">(</span><span class="va">asthma.snps</span>, <span class="va">snpsH2</span><span class="op">)</span></span>
<span><span class="va">haplo.score</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span> <span class="va">haplo.score</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">haplo.stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/haplo.stats/man/haplo.score.slide.html" class="external-link">haplo.score.slide</a></span><span class="op">(</span><span class="va">asthma.snps</span><span class="op">$</span><span class="va">cc</span>, <span class="va">genoH2</span>,</span>
<span>                                         trait.type<span class="op">=</span><span class="st">"binomial"</span>,</span>
<span>                                         n.slide<span class="op">=</span><span class="va">i</span><span class="op">+</span><span class="fl">3</span>,</span>
<span>                                         simulate<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                                         sim.control<span class="op">=</span><span class="fu">haplo.stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/haplo.stats/man/score.sim.control.html" class="external-link">score.sim.control</a></span><span class="op">(</span>min.sim<span class="op">=</span><span class="fl">100</span>,</span>
<span>                                         max.sim<span class="op">=</span><span class="fl">200</span><span class="op">)</span><span class="op">)</span> </span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">haplo.score</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/graphics/title.html" class="external-link">title</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Sliding Window="</span>, <span class="va">i</span><span class="op">+</span><span class="fl">3</span>, sep<span class="op">=</span><span class="st">""</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="gaawr2%2Fhaplo.stats-1.png" width="672" style="display:block; margin: auto"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">snpsH3</span> <span class="op">&lt;-</span> <span class="va">snpsH2</span><span class="op">[</span><span class="fl">4</span><span class="op">:</span><span class="fl">7</span><span class="op">]</span></span>
<span><span class="va">genoH3</span> <span class="op">&lt;-</span> <span class="fu">make.geno</span><span class="op">(</span><span class="va">asthma.snps</span>, <span class="va">snpsH3</span><span class="op">)</span></span>
<span><span class="va">mod</span> <span class="op">&lt;-</span> <span class="fu">haplo.stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/haplo.stats/man/haplo.glm.html" class="external-link">haplo.glm</a></span><span class="op">(</span><span class="va">cc</span><span class="op">~</span><span class="va">genoH3</span>, data<span class="op">=</span><span class="va">asthma.snps</span>,</span>
<span>             family<span class="op">=</span><span class="st">"binomial"</span>,</span>
<span>             locus.label<span class="op">=</span><span class="va">snpsH3</span>,</span>
<span>             allele.lev<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">genoH3</span><span class="op">)</span><span class="op">$</span><span class="va">unique.alleles</span>,</span>
<span>             control <span class="op">=</span> <span class="fu">haplo.stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/haplo.stats/man/haplo.glm.control.html" class="external-link">haplo.glm.control</a></span><span class="op">(</span>haplo.freq.min<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mod</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Call:  haplo.stats::haplo.glm(formula = cc ~ genoH3, family = "binomial", </span></span>
<span><span class="co">#&gt;     data = asthma.snps, locus.label = snpsH3, control = haplo.stats::haplo.glm.control(haplo.freq.min = 0.05), </span></span>
<span><span class="co">#&gt;     allele.lev = attributes(genoH3)$unique.alleles)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Coefficients:</span></span>
<span><span class="co">#&gt; (Intercept)     genoH3.2     genoH3.9    genoH3.11  genoH3.rare  </span></span>
<span><span class="co">#&gt;  -1.2880793    0.0067873   -0.1146385    0.1099134   -0.2505565  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Haplotypes:</span></span>
<span><span class="co">#&gt;             rs1430090 rs6737251 rs11685217 rs1430097 hap.freq</span></span>
<span><span class="co">#&gt; genoH3.2            G         C          C         C  0.29632</span></span>
<span><span class="co">#&gt; genoH3.9            T         T          C         A  0.11839</span></span>
<span><span class="co">#&gt; genoH3.11           T         T          T         A  0.18302</span></span>
<span><span class="co">#&gt; genoH3.rare         *         *          *         *  0.05021</span></span>
<span><span class="co">#&gt; haplo.base          T         C          C         C  0.35206</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Degrees of Freedom:  1577 Total (i.e. Null);  1573 Residual</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Null Deviance:  1645 </span></span>
<span><span class="co">#&gt;  Residual Deviance:  1641 </span></span>
<span><span class="co">#&gt;                AIC:  1651</span></span>
<span><span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/intervals.html" class="external-link">intervals</a></span><span class="op">(</span><span class="va">mod</span><span class="op">)</span></span>
<span><span class="co">#&gt;                freq   or   95%   C.I.   P-val </span></span>
<span><span class="co">#&gt;         TCCC 0.3521   1.00 Reference haplotype </span></span>
<span><span class="co">#&gt;         GCCC 0.2963   1.01 (   0.82 -   1.24 )  0.9495 </span></span>
<span><span class="co">#&gt;         TTCA 0.1184   0.89 (   0.67 -   1.20 )  0.4433 </span></span>
<span><span class="co">#&gt;         TTTA 0.1830   1.12 (   0.87 -   1.43 )  0.3796 </span></span>
<span><span class="co">#&gt;  genoH3.rare 0.0502   0.78 (   0.50 -   1.22 )  0.2750</span></span>
<span><span class="va">mod.adj.ref</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/glm.html" class="external-link">glm</a></span><span class="op">(</span><span class="va">cc</span> <span class="op">~</span> <span class="va">smoke</span>, data<span class="op">=</span><span class="va">asthma.snps</span>, family<span class="op">=</span><span class="st">"binomial"</span><span class="op">)</span></span>
<span><span class="va">mod.adj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/haplo.stats/man/haplo.glm.html" class="external-link">haplo.glm</a></span><span class="op">(</span><span class="va">cc</span> <span class="op">~</span> <span class="va">genoH3</span> <span class="op">+</span> <span class="va">smoke</span>, data<span class="op">=</span><span class="va">asthma.snps</span>,</span>
<span>                 family<span class="op">=</span><span class="st">"binomial"</span>,</span>
<span>                 locus.label<span class="op">=</span><span class="va">snpsH3</span>,</span>
<span>                 allele.lev<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/attributes.html" class="external-link">attributes</a></span><span class="op">(</span><span class="va">genoH3</span><span class="op">)</span><span class="op">$</span><span class="va">unique.alleles</span>,</span>
<span>                 control <span class="op">=</span> <span class="fu">haplo.stats</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/haplo.stats/man/haplo.glm.control.html" class="external-link">haplo.glm.control</a></span><span class="op">(</span>haplo.freq.min<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mod.adj</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Call:  haplo.glm(formula = cc ~ genoH3 + smoke, family = "binomial", </span></span>
<span><span class="co">#&gt;     data = asthma.snps, locus.label = snpsH3, control = haplo.stats::haplo.glm.control(haplo.freq.min = 0.05), </span></span>
<span><span class="co">#&gt;     allele.lev = attributes(genoH3)$unique.alleles)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Coefficients:</span></span>
<span><span class="co">#&gt; (Intercept)     genoH3.2     genoH3.9    genoH3.11  genoH3.rare        smoke  </span></span>
<span><span class="co">#&gt;   -1.189508     0.019374    -0.097111     0.113149    -0.253046    -0.393112  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Haplotypes:</span></span>
<span><span class="co">#&gt;             rs1430090 rs6737251 rs11685217 rs1430097 hap.freq</span></span>
<span><span class="co">#&gt; genoH3.2            G         C          C         C  0.29539</span></span>
<span><span class="co">#&gt; genoH3.9            T         T          C         A  0.11860</span></span>
<span><span class="co">#&gt; genoH3.11           T         T          T         A  0.18319</span></span>
<span><span class="co">#&gt; genoH3.rare         *         *          *         *  0.05044</span></span>
<span><span class="co">#&gt; haplo.base          T         C          C         C  0.35238</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Degrees of Freedom:  1570 Total (i.e. Null);  1565 Residual</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Subjects removed by NAs in y or x, or all NA in geno</span></span>
<span><span class="co">#&gt;   yxmiss genomiss </span></span>
<span><span class="co">#&gt;        7        0 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Null Deviance:  1639 </span></span>
<span><span class="co">#&gt;  Residual Deviance:  1627 </span></span>
<span><span class="co">#&gt;                AIC:  1639</span></span>
<span><span class="va">lrt.adj</span> <span class="op">&lt;-</span> <span class="va">mod.adj.ref</span><span class="op">$</span><span class="va">deviance</span> <span class="op">-</span> <span class="va">mod.adj</span><span class="op">$</span><span class="va">deviance</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/Chisquare.html" class="external-link">pchisq</a></span><span class="op">(</span><span class="va">lrt.adj</span>, <span class="va">mod.adj</span><span class="op">$</span><span class="va">lrt</span><span class="op">$</span><span class="va">df</span>, lower<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.65</span></span></code></pre></div>
</div>
<div class="section level3" number="3.4">
<h3 id="gmmat">
<span class="header-section-number">3.4</span> GMMAT<a class="anchor" aria-label="anchor" href="#gmmat"></a>
</h3>
<p>The following is modified slightly from the package vignette,</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">example</span>,package<span class="op">=</span><span class="st">"GMMAT"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/attach.html" class="external-link">attach</a></span><span class="op">(</span><span class="va">example</span><span class="op">)</span></span>
<span><span class="va">model0</span> <span class="op">&lt;-</span> <span class="fu">GMMAT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GMMAT/man/glmmkin.html" class="external-link">glmmkin</a></span><span class="op">(</span><span class="va">disease</span> <span class="op">~</span> <span class="va">age</span> <span class="op">+</span> <span class="va">sex</span>, data <span class="op">=</span> <span class="va">pheno</span>, kins <span class="op">=</span> <span class="va">GRM</span>,</span>
<span>                         id <span class="op">=</span> <span class="st">"id"</span>, family <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/family.html" class="external-link">binomial</a></span><span class="op">(</span>link <span class="op">=</span> <span class="st">"logit"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">model1</span> <span class="op">&lt;-</span> <span class="fu">GMMAT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GMMAT/man/glmmkin.html" class="external-link">glmmkin</a></span><span class="op">(</span>fixed <span class="op">=</span> <span class="va">trait</span> <span class="op">~</span> <span class="va">age</span> <span class="op">+</span> <span class="va">sex</span>, data <span class="op">=</span> <span class="va">pheno</span>, kins <span class="op">=</span> <span class="va">GRM</span>,</span>
<span>                         id <span class="op">=</span> <span class="st">"id"</span>, family <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/family.html" class="external-link">gaussian</a></span><span class="op">(</span>link <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">model2</span> <span class="op">&lt;-</span> <span class="fu">GMMAT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GMMAT/man/glmmkin.html" class="external-link">glmmkin</a></span><span class="op">(</span>fixed <span class="op">=</span> <span class="va">trait</span> <span class="op">~</span> <span class="va">age</span> <span class="op">+</span> <span class="va">sex</span>, data <span class="op">=</span> <span class="va">pheno</span>, kins <span class="op">=</span> <span class="va">GRM</span>,</span>
<span>                         id <span class="op">=</span> <span class="st">"id"</span>, groups <span class="op">=</span> <span class="st">"disease"</span>,</span>
<span>                         family <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/family.html" class="external-link">gaussian</a></span><span class="op">(</span>link <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">snps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SNP10"</span>, <span class="st">"SNP25"</span>, <span class="st">"SNP1"</span>, <span class="st">"SNP0"</span><span class="op">)</span></span>
<span><span class="va">geno.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"geno.bgen"</span>, package <span class="op">=</span> <span class="st">"GMMAT"</span><span class="op">)</span></span>
<span><span class="va">samplefile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"geno.sample"</span>, package <span class="op">=</span> <span class="st">"GMMAT"</span><span class="op">)</span></span>
<span><span class="va">outfile</span> <span class="op">&lt;-</span> <span class="st">"glmm.score.txt"</span></span>
<span><span class="fu">GMMAT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GMMAT/man/glmm.score.html" class="external-link">glmm.score</a></span><span class="op">(</span><span class="va">model0</span>, infile <span class="op">=</span> <span class="va">geno.file</span>, BGEN.samplefile <span class="op">=</span> <span class="va">samplefile</span>,</span>
<span>                  outfile <span class="op">=</span> <span class="va">outfile</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span><span class="va">outfile</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>     <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span>n<span class="op">=</span><span class="fl">4</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>     <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption<span class="op">=</span><span class="st">"Score tests under GLMM on four SNPs"</span>,digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>
<span id="tab:gmmat">Table 3.3: </span>Score tests under GLMM on four SNPs</caption>
<thead><tr class="header">
<th align="left">SNP</th>
<th align="left">RSID</th>
<th align="right">CHR</th>
<th align="right">POS</th>
<th align="left">A1</th>
<th align="left">A2</th>
<th align="right">N</th>
<th align="right">AF</th>
<th align="right">SCORE</th>
<th align="right">VAR</th>
<th align="right">PVAL</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">SNP1</td>
<td align="left">SNP1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="left">T</td>
<td align="left">A</td>
<td align="right">393</td>
<td align="right">0.97</td>
<td align="right">-1.99</td>
<td align="right">4.6</td>
<td align="right">0.35</td>
</tr>
<tr class="even">
<td align="left">SNP2</td>
<td align="left">SNP2</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="left">A</td>
<td align="left">C</td>
<td align="right">400</td>
<td align="right">0.50</td>
<td align="right">3.51</td>
<td align="right">46.3</td>
<td align="right">0.61</td>
</tr>
<tr class="odd">
<td align="left">SNP3</td>
<td align="left">SNP3</td>
<td align="right">1</td>
<td align="right">3</td>
<td align="left">C</td>
<td align="left">A</td>
<td align="right">400</td>
<td align="right">0.79</td>
<td align="right">0.53</td>
<td align="right">30.6</td>
<td align="right">0.92</td>
</tr>
<tr class="even">
<td align="left">SNP4</td>
<td align="left">SNP4</td>
<td align="right">1</td>
<td align="right">4</td>
<td align="left">G</td>
<td align="left">A</td>
<td align="right">400</td>
<td align="right">0.70</td>
<td align="right">3.11</td>
<td align="right">40.5</td>
<td align="right">0.62</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">outfile</span><span class="op">)</span></span>
<span><span class="va">bed.file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"geno.bed"</span>, package <span class="op">=</span> <span class="st">"GMMAT"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>            <span class="fu">tools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/tools/fileutils.html" class="external-link">file_path_sans_ext</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">model.wald</span> <span class="op">&lt;-</span> <span class="fu">GMMAT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GMMAT/man/glmm.wald.html" class="external-link">glmm.wald</a></span><span class="op">(</span>fixed <span class="op">=</span> <span class="va">disease</span> <span class="op">~</span> <span class="va">age</span> <span class="op">+</span> <span class="va">sex</span>, data <span class="op">=</span> <span class="va">pheno</span>,</span>
<span>                               kins <span class="op">=</span> <span class="va">GRM</span>, id <span class="op">=</span> <span class="st">"id"</span>, family <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/family.html" class="external-link">binomial</a></span><span class="op">(</span>link <span class="op">=</span> <span class="st">"logit"</span><span class="op">)</span>,</span>
<span>                               infile <span class="op">=</span> <span class="va">bed.file</span>, snps <span class="op">=</span> <span class="va">snps</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">model.wald</span>,caption<span class="op">=</span><span class="st">"Wald tests under GLMM on four SNPs"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>
<span id="tab:gmmat">Table 3.3: </span>Wald tests under GLMM on four SNPs</caption>
<thead><tr class="header">
<th align="left">CHR</th>
<th align="left">SNP</th>
<th align="left">cM</th>
<th align="left">POS</th>
<th align="left">A1</th>
<th align="left">A2</th>
<th align="right">N</th>
<th align="right">AF</th>
<th align="right">BETA</th>
<th align="right">SE</th>
<th align="right">PVAL</th>
<th align="left">converged</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="left">SNP10</td>
<td align="left">0</td>
<td align="left">10</td>
<td align="left">G</td>
<td align="left">A</td>
<td align="right">400</td>
<td align="right">0.77</td>
<td align="right">0.14</td>
<td align="right">0.17</td>
<td align="right">0.42</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">1</td>
<td align="left">SNP25</td>
<td align="left">0</td>
<td align="left">25</td>
<td align="left">C</td>
<td align="left">A</td>
<td align="right">400</td>
<td align="right">0.82</td>
<td align="right">0.03</td>
<td align="right">0.19</td>
<td align="right">0.88</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">1</td>
<td align="left">SNP1</td>
<td align="left">0</td>
<td align="left">1</td>
<td align="left">T</td>
<td align="left">A</td>
<td align="right">393</td>
<td align="right">0.97</td>
<td align="right">-0.46</td>
<td align="right">0.49</td>
<td align="right">0.35</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">NA</td>
<td align="left">SNP0</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/detach.html" class="external-link">detach</a></span><span class="op">(</span><span class="va">example</span><span class="op">)</span></span></code></pre></div>
<p>where both BGEN and PLINK binary file are illustrated.</p>
</div>
<div class="section level3" number="3.5">
<h3 id="proteogenomics">
<span class="header-section-number">3.5</span> Proteogenomics<a class="anchor" aria-label="anchor" href="#proteogenomics"></a>
</h3>
<p>See <a href="https://jinghuazhao.github.io/pQTLtools/" class="external-link uri">https://jinghuazhao.github.io/pQTLtools/</a>. An overview on the proteomics is found from the relevant paper<span class="citation"><sup>1</sup></span>.</p>
</div>
<div class="section level3" number="3.6">
<h3 id="gwas">
<span class="header-section-number">3.6</span> GWAS<a class="anchor" aria-label="anchor" href="#gwas"></a>
</h3>
<p>We return to the asthma data used in <strong>SNPassoc</strong>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">assoc</span> <span class="op">&lt;-</span> <span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/WGassociation.html" class="external-link">WGassociation</a></span><span class="op">(</span><span class="va">cc</span>, data<span class="op">=</span><span class="va">asthma.snps</span><span class="op">)</span></span>
<span><span class="va">assoc.adj</span> <span class="op">&lt;-</span> <span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/WGassociation.html" class="external-link">WGassociation</a></span><span class="op">(</span><span class="va">cc</span> <span class="op">~</span> <span class="va">country</span> <span class="op">+</span> <span class="va">smoke</span>, <span class="va">asthma.snps</span><span class="op">)</span></span>
<span><span class="va">assoc.maxstat</span> <span class="op">&lt;-</span> <span class="fu">SNPassoc</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SNPassoc/man/maxstat.html" class="external-link">maxstat</a></span><span class="op">(</span><span class="va">asthma.snps</span>, <span class="va">cc</span><span class="op">)</span></span>
<span><span class="va">assoc</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">comments</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption<span class="op">=</span><span class="st">"SNP association"</span><span class="op">)</span></span>
<span><span class="va">assoc.adj</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">comments</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption<span class="op">=</span><span class="st">"with adjustment for contountry &amp; smoking"</span><span class="op">)</span></span>
<span><span class="va">assoc.maxstat</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">`[`</span><span class="op">(</span>,<span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"Max stat association statistics"</span><span class="op">)</span></span></code></pre></div>
<p>where assoc.maxstat is coerced into a matrix later, but there appears problematic to knit though.</p>
</div>
<div class="section level3" number="3.7">
<h3 id="h2-jags">
<span class="header-section-number">3.7</span> h2.jags<a class="anchor" aria-label="anchor" href="#h2-jags"></a>
</h3>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1234567</span><span class="op">)</span></span>
<span><span class="va">meyer</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">within</a></span><span class="op">(</span><span class="va">meyer</span>,<span class="op">{</span></span>
<span>         <span class="va">y</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">rnorm</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">y</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">y</span>,na.rm<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/stats/sd.html" class="external-link">sd</a></span><span class="op">(</span><span class="va">y</span>,na.rm<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>         <span class="va">g1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">generation</span><span class="op">==</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">0</span><span class="op">)</span></span>
<span>         <span class="va">g2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">generation</span><span class="op">==</span><span class="fl">2</span>,<span class="fl">1</span>,<span class="fl">0</span><span class="op">)</span></span>
<span>         <span class="va">id</span> <span class="op">&lt;-</span> <span class="va">animal</span></span>
<span>         <span class="va">animal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">animal</span><span class="op">)</span>,<span class="va">animal</span>,<span class="fl">0</span><span class="op">)</span></span>
<span>         <span class="va">dam</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">dam</span><span class="op">)</span>,<span class="va">dam</span>,<span class="fl">0</span><span class="op">)</span></span>
<span>         <span class="va">sire</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">sire</span><span class="op">)</span>,<span class="va">sire</span>,<span class="fl">0</span><span class="op">)</span></span>
<span>     <span class="op">}</span><span class="op">)</span></span>
<span><span class="va">G</span> <span class="op">&lt;-</span> <span class="fu">gap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gap/man/kin.morgan.html" class="external-link">kin.morgan</a></span><span class="op">(</span><span class="va">meyer</span><span class="op">)</span><span class="op">$</span><span class="va">kin.matrix</span><span class="op">*</span><span class="fl">2</span></span>
<span><span class="va">r</span> <span class="op">&lt;-</span> <span class="fu">regress</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/regress/man/regress.html" class="external-link">regress</a></span><span class="op">(</span><span class="va">y</span><span class="op">~</span><span class="op">-</span><span class="fl">1</span><span class="op">+</span><span class="va">g1</span><span class="op">+</span><span class="va">g2</span>,<span class="op">~</span><span class="va">G</span>,data<span class="op">=</span><span class="va">meyer</span><span class="op">)</span></span>
<span><span class="va">r</span></span>
<span><span class="co">#&gt; Likelihood kernel: K = g1+g2</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Maximized log likelihood with kernel K is  -844 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Linear Coefficients:</span></span>
<span><span class="co">#&gt;     Estimate Std. Error</span></span>
<span><span class="co">#&gt;  g1      223        1.4</span></span>
<span><span class="co">#&gt;  g2      239        1.8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Variance Coefficients:</span></span>
<span><span class="co">#&gt;     Estimate Std. Error</span></span>
<span><span class="co">#&gt;  G        32         14</span></span>
<span><span class="co">#&gt;  In       72         10</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">r</span>,<span class="fu">gap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gap/man/h2G.html" class="external-link">h2G</a></span><span class="op">(</span><span class="va">sigma</span>,<span class="va">sigma.cov</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Vp = 104 SE = 9.9 </span></span>
<span><span class="co">#&gt; h2G = 0.3 SE = 0.11</span></span>
<span><span class="va">eps</span> <span class="op">&lt;-</span> <span class="fl">0.001</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">meyer</span>,<span class="va">y</span><span class="op">)</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">meyer</span>,<span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">g1</span>,<span class="va">g2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ex</span> <span class="op">&lt;-</span> <span class="fu">gap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gap/man/h2.jags.html" class="external-link">h2.jags</a></span><span class="op">(</span><span class="va">y</span>,<span class="va">x</span>,<span class="va">G</span>,sigma.p<span class="op">=</span><span class="fl">0.03</span>,sigma.r<span class="op">=</span><span class="fl">0.014</span><span class="op">)</span></span>
<span><span class="co">#&gt; Compiling model graph</span></span>
<span><span class="co">#&gt;    Resolving undeclared variables</span></span>
<span><span class="co">#&gt;    Allocating nodes</span></span>
<span><span class="co">#&gt; Graph information:</span></span>
<span><span class="co">#&gt;    Observed stochastic nodes: 306</span></span>
<span><span class="co">#&gt;    Unobserved stochastic nodes: 310</span></span>
<span><span class="co">#&gt;    Total graph size: 95805</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Initializing model</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">ex</span><span class="op">$</span><span class="va">BUGSoutput</span><span class="op">$</span><span class="va">summary</span>,digits<span class="op">=</span><span class="fl">2</span>,caption<span class="op">=</span><span class="st">"MCMC results for the Meyer data"</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<caption>
<span id="tab:jags">Table 3.4: </span>MCMC results for the Meyer data</caption>
<colgroup>
<col width="12%">
<col width="9%">
<col width="8%">
<col width="11%">
<col width="11%">
<col width="9%">
<col width="11%">
<col width="11%">
<col width="6%">
<col width="8%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">mean</th>
<th align="right">sd</th>
<th align="right">2.5%</th>
<th align="right">25%</th>
<th align="right">50%</th>
<th align="right">75%</th>
<th align="right">97.5%</th>
<th align="right">Rhat</th>
<th align="right">n.eff</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">b[1]</td>
<td align="right">222.2</td>
<td align="right">1.45</td>
<td align="right">219.35</td>
<td align="right">221.14</td>
<td align="right">222.2</td>
<td align="right">223.10</td>
<td align="right">225.06</td>
<td align="right">1</td>
<td align="right">3000</td>
</tr>
<tr class="even">
<td align="left">b[2]</td>
<td align="right">237.4</td>
<td align="right">1.77</td>
<td align="right">233.89</td>
<td align="right">236.27</td>
<td align="right">237.5</td>
<td align="right">238.59</td>
<td align="right">240.90</td>
<td align="right">1</td>
<td align="right">2300</td>
</tr>
<tr class="odd">
<td align="left">deviance</td>
<td align="right">2180.9</td>
<td align="right">26.86</td>
<td align="right">2120.86</td>
<td align="right">2165.26</td>
<td align="right">2182.6</td>
<td align="right">2198.78</td>
<td align="right">2228.85</td>
<td align="right">1</td>
<td align="right">140</td>
</tr>
<tr class="even">
<td align="left">h2</td>
<td align="right">0.3</td>
<td align="right">0.09</td>
<td align="right">0.14</td>
<td align="right">0.25</td>
<td align="right">0.3</td>
<td align="right">0.36</td>
<td align="right">0.48</td>
<td align="right">1</td>
<td align="right">120</td>
</tr>
<tr class="odd">
<td align="left">p</td>
<td align="right">32.7</td>
<td align="right">10.96</td>
<td align="right">14.26</td>
<td align="right">25.17</td>
<td align="right">31.9</td>
<td align="right">38.61</td>
<td align="right">56.92</td>
<td align="right">1</td>
<td align="right">120</td>
</tr>
<tr class="even">
<td align="left">r</td>
<td align="right">73.7</td>
<td align="right">8.72</td>
<td align="right">57.30</td>
<td align="right">67.77</td>
<td align="right">73.5</td>
<td align="right">79.29</td>
<td align="right">91.18</td>
<td align="right">1</td>
<td align="right">170</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3" number="3.8">
<h3 id="powereqtl">
<span class="header-section-number">3.8</span> powerEQTL<a class="anchor" aria-label="anchor" href="#powereqtl"></a>
</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">n.designs</span> <span class="op">&lt;-</span> <span class="fl">6</span></span>
<span><span class="va">designs</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="va">n.designs</span></span>
<span><span class="va">N</span> <span class="op">&lt;-</span> <span class="fl">50</span> <span class="op">*</span> <span class="va">designs</span></span>
<span><span class="va">n.grids</span> <span class="op">&lt;-</span> <span class="fl">100</span></span>
<span><span class="va">index</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="va">n.grids</span></span>
<span><span class="va">grids</span> <span class="op">&lt;-</span> <span class="va">index</span> <span class="op">/</span> <span class="va">n.grids</span></span>
<span><span class="va">MAF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0.005</span>, <span class="va">n.grids</span><span class="op">/</span><span class="fl">2</span>, by<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span> <span class="op">/</span> <span class="va">n.grids</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">MAF</span>,<span class="va">grids</span>,type<span class="op">=</span><span class="st">"n"</span>,ylab<span class="op">=</span><span class="st">"Power"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html" class="external-link">mtext</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/expression.html" class="external-link">expression</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"("</span>,<span class="va">alpha</span>,<span class="st">" = 0.05)"</span><span class="op">)</span><span class="op">)</span>,<span class="fl">1</span>,line<span class="op">=</span><span class="fl">4.5</span><span class="op">)</span></span>
<span><span class="va">colors</span> <span class="op">&lt;-</span> <span class="fu">grDevices</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/grDevices/palettes.html" class="external-link">hcl.colors</a></span><span class="op">(</span><span class="va">n.designs</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">design</span> <span class="kw">in</span> <span class="va">designs</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="va">power.SLR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="cn">NA</span>,<span class="va">n.grids</span><span class="op">)</span></span>
<span>  <span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="va">index</span><span class="op">)</span> <span class="va">power.SLR</span><span class="op">[</span><span class="va">j</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">powerEQTL</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/powerEQTL/man/powerEQTL.SLR.html" class="external-link">powerEQTL.SLR</a></span><span class="op">(</span>MAF <span class="op">=</span> <span class="va">MAF</span><span class="op">[</span><span class="va">j</span><span class="op">]</span>, FWER <span class="op">=</span> <span class="fl">0.05</span>, nTests <span class="op">=</span> <span class="fl">240</span>, slope <span class="op">=</span> <span class="fl">0.13</span>,</span>
<span>                                                            n <span class="op">=</span> <span class="va">N</span><span class="op">[</span><span class="va">design</span><span class="op">]</span>, sigma.y <span class="op">=</span> <span class="fl">0.13</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="va">MAF</span>,<span class="va">power.SLR</span>,col<span class="op">=</span><span class="va">colors</span><span class="op">[</span><span class="va">design</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span><span class="st">"bottomright"</span>, inset<span class="op">=</span><span class="fl">.02</span>, title<span class="op">=</span><span class="st">"Sample size (N)"</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">N</span><span class="op">)</span>, col<span class="op">=</span><span class="va">colors</span>, horiz<span class="op">=</span><span class="cn">FALSE</span>, cex<span class="op">=</span><span class="fl">0.8</span>, lty<span class="op">=</span><span class="va">designs</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:eqtl"></span>
<img src="gaawr2%2Feqtl-1.png" alt="Power Estimation for eQTL Studies of 240 SNPs" width="768" style="display:block; margin: auto"><p class="caption">
Figure 3.2: Power Estimation for eQTL Studies of 240 SNPs
</p>
</div>
<div class="section level4" number="3.8.1">
<h4 id="opentargets">
<span class="header-section-number">3.8.1</span> OpenTargets<a class="anchor" aria-label="anchor" href="#opentargets"></a>
</h4>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_id</span> <span class="op">&lt;-</span> <span class="st">"ENSG00000164308"</span></span>
<span><span class="va">query_string</span> <span class="op">=</span> <span class="st">"</span></span>
<span><span class="st">  query target($ensemblId: String!){</span></span>
<span><span class="st">    target(ensemblId: $ensemblId){</span></span>
<span><span class="st">      id</span></span>
<span><span class="st">      approvedSymbol</span></span>
<span><span class="st">      biotype</span></span>
<span><span class="st">      geneticConstraint {</span></span>
<span><span class="st">        constraintType</span></span>
<span><span class="st">        exp</span></span>
<span><span class="st">        obs</span></span>
<span><span class="st">        score</span></span>
<span><span class="st">        oe</span></span>
<span><span class="st">        oeLower</span></span>
<span><span class="st">        oeUpper</span></span>
<span><span class="st">      }</span></span>
<span><span class="st">      tractability {</span></span>
<span><span class="st">        label</span></span>
<span><span class="st">        modality</span></span>
<span><span class="st">        value</span></span>
<span><span class="st">      }</span></span>
<span><span class="st">    }</span></span>
<span><span class="st">  }</span></span>
<span><span class="st">"</span></span>
<span><span class="va">base_url</span> <span class="op">&lt;-</span> <span class="st">"https://api.platform.opentargets.org/api/v4/graphql"</span></span>
<span><span class="va">variables</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"ensemblId"</span> <span class="op">=</span> <span class="va">gene_id</span><span class="op">)</span></span>
<span><span class="va">post_body</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">query_string</span>, variables <span class="op">=</span> <span class="va">variables</span><span class="op">)</span></span>
<span><span class="va">r</span> <span class="op">&lt;-</span> <span class="fu">httr</span><span class="fu">::</span><span class="fu"><a href="https://httr.r-lib.org/reference/POST.html" class="external-link">POST</a></span><span class="op">(</span>url<span class="op">=</span><span class="va">base_url</span>, body<span class="op">=</span><span class="va">post_body</span>, encode<span class="op">=</span><span class="st">'json'</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/iconv.html" class="external-link">iconv</a></span><span class="op">(</span><span class="va">r</span>, <span class="st">"latin1"</span>, <span class="st">"ASCII"</span><span class="op">)</span></span>
<span><span class="co">#&gt; No encoding supplied: defaulting to UTF-8.</span></span>
<span><span class="va">content</span> <span class="op">&lt;-</span> <span class="fu">jsonlite</span><span class="fu">::</span><span class="fu"><a href="https://jeroen.r-universe.dev/jsonlite/reference/fromJSON.html" class="external-link">fromJSON</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">content</span>, caption<span class="op">=</span><span class="st">"Query of ERAP2 from OpenTargets"</span><span class="op">)</span></span></code></pre></div>
<table class="table kable_wrapper">
<caption>
<span id="tab:unnamed-chunk-2">Table 3.5: </span>Query of ERAP2 from OpenTargets
</caption>
<tbody><tr>
<td>
<table class="table kable_wrapper"><tbody><tr>
<td>
<table class="table kable_wrapper"><tbody><tr>
<td>
<table class="table">
<thead><tr class="header">
<th align="left">x</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">ENSG00000164308</td>
</tr></tbody>
</table>
</td>
<td>
<table class="table">
<thead><tr class="header">
<th align="left">x</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">ERAP2</td>
</tr></tbody>
</table>
</td>
<td>
<table class="table">
<thead><tr class="header">
<th align="left">x</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">protein_coding</td>
</tr></tbody>
</table>
</td>
<td>
<table class="table">
<thead><tr class="header">
<th align="left">constraintType</th>
<th align="right">exp</th>
<th align="right">obs</th>
<th align="right">score</th>
<th align="right">oe</th>
<th align="right">oeLower</th>
<th align="right">oeUpper</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">syn</td>
<td align="right">186</td>
<td align="right">183</td>
<td align="right">0.15</td>
<td align="right">0.99</td>
<td align="right">0.87</td>
<td align="right">1.1</td>
</tr>
<tr class="even">
<td align="left">mis</td>
<td align="right">497</td>
<td align="right">501</td>
<td align="right">-0.07</td>
<td align="right">1.01</td>
<td align="right">0.94</td>
<td align="right">1.1</td>
</tr>
<tr class="odd">
<td align="left">lof</td>
<td align="right">48</td>
<td align="right">45</td>
<td align="right">0.00</td>
<td align="right">0.93</td>
<td align="right">0.73</td>
<td align="right">1.2</td>
</tr>
</tbody>
</table>
</td>
<td>
<table class="table">
<thead><tr class="header">
<th align="left">label</th>
<th align="left">modality</th>
<th align="left">value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Approved Drug</td>
<td align="left">SM</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">Advanced Clinical</td>
<td align="left">SM</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">Phase 1 Clinical</td>
<td align="left">SM</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">Structure with Ligand</td>
<td align="left">SM</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">High-Quality Ligand</td>
<td align="left">SM</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">High-Quality Pocket</td>
<td align="left">SM</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">Med-Quality Pocket</td>
<td align="left">SM</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">Druggable Family</td>
<td align="left">SM</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">Approved Drug</td>
<td align="left">AB</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">Advanced Clinical</td>
<td align="left">AB</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">Phase 1 Clinical</td>
<td align="left">AB</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">UniProt loc high conf</td>
<td align="left">AB</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">GO CC high conf</td>
<td align="left">AB</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">UniProt loc med conf</td>
<td align="left">AB</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">UniProt SigP or TMHMM</td>
<td align="left">AB</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">GO CC med conf</td>
<td align="left">AB</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">Human Protein Atlas loc</td>
<td align="left">AB</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">Approved Drug</td>
<td align="left">PR</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">Advanced Clinical</td>
<td align="left">PR</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">Phase 1 Clinical</td>
<td align="left">PR</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">Literature</td>
<td align="left">PR</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">UniProt Ubiquitination</td>
<td align="left">PR</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">Database Ubiquitination</td>
<td align="left">PR</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">Half-life Data</td>
<td align="left">PR</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">Small Molecule Binder</td>
<td align="left">PR</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">Approved Drug</td>
<td align="left">OC</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">Advanced Clinical</td>
<td align="left">OC</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">Phase 1 Clinical</td>
<td align="left">OC</td>
<td align="left">FALSE</td>
</tr>
</tbody>
</table>
</td>
</tr></tbody></table>
</td>
</tr></tbody></table>
</td>
</tr></tbody>
</table>
</div>
</div>
</div>
<div class="section level2" number="4">
<h2 id="annotations">
<span class="header-section-number">4</span> Annotations<a class="anchor" aria-label="anchor" href="#annotations"></a>
</h2>
<div class="section level3" number="4.1">
<h3 id="ensdb-hsapiens-v75">
<span class="header-section-number">4.1</span> EnsDb.Hsapiens.v75<a class="anchor" aria-label="anchor" href="#ensdb-hsapiens-v75"></a>
</h3>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">ensembldb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">EnsDb.Hsapiens.v75</span><span class="op">)</span></span>
<span><span class="co">#&gt;                  name                               value</span></span>
<span><span class="co">#&gt; 1             Db type                               EnsDb</span></span>
<span><span class="co">#&gt; 2     Type of Gene ID                     Ensembl Gene ID</span></span>
<span><span class="co">#&gt; 3  Supporting package                           ensembldb</span></span>
<span><span class="co">#&gt; 4       Db created by ensembldb package from Bioconductor</span></span>
<span><span class="co">#&gt; 5      script_version                               0.3.0</span></span>
<span><span class="co">#&gt; 6       Creation time            Thu May 18 09:15:45 2017</span></span>
<span><span class="co">#&gt; 7     ensembl_version                                  75</span></span>
<span><span class="co">#&gt; 8        ensembl_host                           localhost</span></span>
<span><span class="co">#&gt; 9            Organism                        homo_sapiens</span></span>
<span><span class="co">#&gt; 10        taxonomy_id                                9606</span></span>
<span><span class="co">#&gt; 11       genome_build                              GRCh37</span></span>
<span><span class="co">#&gt; 12    DBSCHEMAVERSION                                 2.0</span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu">ensembldb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">EnsDb.Hsapiens.v75</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRanges object with 6 ranges and 6 metadata columns:</span></span>
<span><span class="co">#&gt;                   seqnames      ranges strand |         gene_id   gene_name gene_biotype seq_coord_system      symbol                          entrezid</span></span>
<span><span class="co">#&gt;                      &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt; &lt;character&gt;  &lt;character&gt;      &lt;character&gt; &lt;character&gt;                            &lt;list&gt;</span></span>
<span><span class="co">#&gt;   ENSG00000223972        1 11869-14412      + | ENSG00000223972     DDX11L1   pseudogene       chromosome     DDX11L1               100287596,100287102</span></span>
<span><span class="co">#&gt;   ENSG00000227232        1 14363-29806      - | ENSG00000227232      WASH7P   pseudogene       chromosome      WASH7P                  100287171,653635</span></span>
<span><span class="co">#&gt;   ENSG00000243485        1 29554-31109      + | ENSG00000243485  MIR1302-10      lincRNA       chromosome  MIR1302-10 100422919,100422834,100422831,...</span></span>
<span><span class="co">#&gt;   ENSG00000237613        1 34554-36081      - | ENSG00000237613     FAM138A      lincRNA       chromosome     FAM138A              654835,645520,641702</span></span>
<span><span class="co">#&gt;   ENSG00000268020        1 52473-54936      + | ENSG00000268020      OR4G4P   pseudogene       chromosome      OR4G4P                              &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   ENSG00000240361        1 62948-63887      + | ENSG00000240361     OR4G11P   pseudogene       chromosome     OR4G11P                              &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 273 sequences (1 circular) from GRCh37 genome</span></span>
<span><span class="va">transcripts_data</span> <span class="op">&lt;-</span> <span class="fu">ensembldb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">transcripts</a></span><span class="op">(</span><span class="va">EnsDb.Hsapiens.v75</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">transcripts_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRanges object with 6 ranges and 6 metadata columns:</span></span>
<span><span class="co">#&gt;                   seqnames      ranges strand |           tx_id             tx_biotype tx_cds_seq_start tx_cds_seq_end         gene_id         tx_name</span></span>
<span><span class="co">#&gt;                      &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;            &lt;character&gt;        &lt;integer&gt;      &lt;integer&gt;     &lt;character&gt;     &lt;character&gt;</span></span>
<span><span class="co">#&gt;   ENST00000456328        1 11869-14409      + | ENST00000456328   processed_transcript             &lt;NA&gt;           &lt;NA&gt; ENSG00000223972 ENST00000456328</span></span>
<span><span class="co">#&gt;   ENST00000515242        1 11872-14412      + | ENST00000515242 transcribed_unproces..             &lt;NA&gt;           &lt;NA&gt; ENSG00000223972 ENST00000515242</span></span>
<span><span class="co">#&gt;   ENST00000518655        1 11874-14409      + | ENST00000518655 transcribed_unproces..             &lt;NA&gt;           &lt;NA&gt; ENSG00000223972 ENST00000518655</span></span>
<span><span class="co">#&gt;   ENST00000450305        1 12010-13670      + | ENST00000450305 transcribed_unproces..             &lt;NA&gt;           &lt;NA&gt; ENSG00000223972 ENST00000450305</span></span>
<span><span class="co">#&gt;   ENST00000438504        1 14363-29370      - | ENST00000438504 unprocessed_pseudogene             &lt;NA&gt;           &lt;NA&gt; ENSG00000227232 ENST00000438504</span></span>
<span><span class="co">#&gt;   ENST00000541675        1 14363-24886      - | ENST00000541675 unprocessed_pseudogene             &lt;NA&gt;           &lt;NA&gt; ENSG00000227232 ENST00000541675</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 273 sequences (1 circular) from GRCh37 genome</span></span></code></pre></div>
<p>One can also use <code>exons_data &lt;- ensembldb::exons(EnsDb.Hsapiens.v75);head(exons_data)</code> but it is skipped for being considerably longer.</p>
</div>
<div class="section level3" number="4.2">
<h3 id="biomart">
<span class="header-section-number">4.2</span> biomaRt<a class="anchor" aria-label="anchor" href="#biomart"></a>
</h3>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/grimbough/biomaRt" class="external-link">biomaRt</a></span><span class="op">)</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu">biomaRt</span><span class="fu">::</span><span class="fu">martBMCheck</span><span class="op">(</span><span class="va">mart</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"The BioMart service is currently unavailable."</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listMarts.html" class="external-link">listMarts</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ensembl</span> <span class="op">&lt;-</span> <span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useMart.html" class="external-link">useMart</a></span><span class="op">(</span><span class="st">"ensembl"</span>, dataset<span class="op">=</span><span class="st">"hsapiens_gene_ensembl"</span>, host<span class="op">=</span><span class="st">"grch37.ensembl.org"</span>, path<span class="op">=</span><span class="st">"/biomart/martservice"</span><span class="op">)</span></span>
<span><span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listDatasets.html" class="external-link">listDatasets</a></span><span class="op">(</span><span class="va">ensembl</span><span class="op">)</span></span>
<span><span class="va">attr</span> <span class="op">&lt;-</span> <span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listAttributes.html" class="external-link">listAttributes</a></span><span class="op">(</span><span class="va">ensembl</span><span class="op">)</span></span>
<span><span class="va">attr_select</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'ensembl_gene_id'</span>, <span class="st">'chromosome_name'</span>, <span class="st">'start_position'</span>, <span class="st">'end_position'</span>, <span class="st">'description'</span>, <span class="st">'hgnc_symbol'</span>,</span>
<span>                 <span class="st">'transcription_start_site'</span><span class="op">)</span></span>
<span><span class="va">gene</span> <span class="op">&lt;-</span> <span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/getBM.html" class="external-link">getBM</a></span><span class="op">(</span>attributes <span class="op">=</span> <span class="va">attr_select</span>, mart <span class="op">=</span> <span class="va">ensembl</span><span class="op">)</span></span>
<span><span class="va">filter</span> <span class="op">&lt;-</span> <span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listFilters.html" class="external-link">listFilters</a></span><span class="op">(</span><span class="va">ensembl</span><span class="op">)</span></span>
<span><span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/listFilters.html" class="external-link">searchFilters</a></span><span class="op">(</span>mart <span class="op">=</span> <span class="va">ensembl</span>, pattern <span class="op">=</span> <span class="st">"gene"</span><span class="op">)</span></span>
<span><span class="co"># GRCh38</span></span>
<span><span class="va">ensembl</span> <span class="op">&lt;-</span> <span class="fu">biomaRt</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/biomaRt/man/useMart.html" class="external-link">useMart</a></span><span class="op">(</span><span class="st">"ensembl"</span>, dataset<span class="op">=</span><span class="st">"hsapiens_gene_ensembl"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="4.3">
<h3 id="experimental-factor-ontology-efo">
<span class="header-section-number">4.3</span> Experimental Factor Ontology (EFO)<a class="anchor" aria-label="anchor" href="#experimental-factor-ontology-efo"></a>
</h3>
<p>The ontology of traits/disease is formally available as this<span class="citation"><sup>2</sup></span>.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ontologyIndex</span><span class="op">)</span></span>
<span><span class="va">id</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">ontology</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="va">inflammatory</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="va">ontology</span><span class="op">$</span><span class="va">name</span>,pattern<span class="op">=</span><span class="st">"inflammatory"</span><span class="op">)</span></span>
<span>  <span class="va">immune</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="va">ontology</span><span class="op">$</span><span class="va">name</span>,pattern<span class="op">=</span><span class="st">"immune"</span><span class="op">)</span></span>
<span>  <span class="va">inf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">union</a></span><span class="op">(</span><span class="va">inflammatory</span>,<span class="va">immune</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>id<span class="op">=</span><span class="va">ontology</span><span class="op">$</span><span class="va">id</span><span class="op">[</span><span class="va">inf</span><span class="op">]</span>,name<span class="op">=</span><span class="va">ontology</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">inf</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co"># GO</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go</span><span class="op">)</span></span>
<span><span class="va">goidname</span> <span class="op">&lt;-</span> <span class="fu">id</span><span class="op">(</span><span class="va">go</span><span class="op">)</span></span>
<span><span class="co"># EFO</span></span>
<span><span class="va">file</span> <span class="op">&lt;-</span> <span class="st">"efo.obo"</span> <span class="co"># efo-3.26.0</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/ontologyIndex/man/get_relation_names.html" class="external-link">get_relation_names</a></span><span class="op">(</span><span class="va">file</span><span class="op">)</span></span>
<span><span class="va">efo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ontologyIndex/man/get_ontology.html" class="external-link">get_ontology</a></span><span class="op">(</span><span class="va">file</span>, extract_tags<span class="op">=</span><span class="st">"everything"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">efo</span><span class="op">)</span> <span class="co"># 89</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">efo</span><span class="op">$</span><span class="va">id</span><span class="op">)</span> <span class="co"># 27962</span></span>
<span><span class="va">efoidname</span> <span class="op">&lt;-</span> <span class="fu">id</span><span class="op">(</span><span class="va">efo</span><span class="op">)</span></span>
<span><span class="va">diseases</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ontologyIndex/man/get_descendants.html" class="external-link">get_descendants</a></span><span class="op">(</span><span class="va">efo</span>,<span class="st">"EFO:0000408"</span><span class="op">)</span></span>
<span><span class="va">efo_0000540</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ontologyIndex/man/get_descendants.html" class="external-link">get_descendants</a></span><span class="op">(</span><span class="va">efo</span>,<span class="st">"EFO:0000540"</span><span class="op">)</span></span>
<span><span class="va">efo_0000540name</span> <span class="op">&lt;-</span> <span class="va">efo</span><span class="op">$</span><span class="va">name</span><span class="op">[</span><span class="va">efo_0000540</span><span class="op">]</span></span>
<span><span class="va">isd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">efo_0000540</span>,<span class="va">efo_0000540name</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/save.html" class="external-link">save</a></span><span class="op">(</span><span class="va">efo</span>,<span class="va">diseases</span>,<span class="va">isd</span>,<span class="va">efoidname</span>,<span class="va">goidname</span>, file<span class="op">=</span><span class="st">"efo.rda"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">isd</span>,file<span class="op">=</span><span class="st">"efo_0000540.csv"</span>,col.names<span class="op">=</span><span class="cn">FALSE</span>,row.names<span class="op">=</span><span class="cn">FALSE</span>,sep<span class="op">=</span><span class="st">","</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ontologyPlot</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/ontologyPlot/man/onto_plot.html" class="external-link">onto_plot</a></span><span class="op">(</span><span class="va">efo</span>,<span class="va">efo_0000540</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3" number="4.4">
<h3 id="rentrez">
<span class="header-section-number">4.4</span> rentrez<a class="anchor" aria-label="anchor" href="#rentrez"></a>
</h3>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/rentrez/" class="external-link">rentrez</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_dbs.html" class="external-link">entrez_dbs</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_db_links.html" class="external-link">entrez_db_links</a></span><span class="op">(</span><span class="st">"pubmed"</span><span class="op">)</span></span>
<span><span class="va">pubmed_fields</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_db_searchable.html" class="external-link">entrez_db_searchable</a></span><span class="op">(</span><span class="st">"pubmed"</span><span class="op">)</span></span>
<span><span class="co"># set_entrez_key("")</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html" class="external-link">Sys.getenv</a></span><span class="op">(</span><span class="st">"ENTREZ_KEY"</span><span class="op">)</span></span>
<span><span class="va">term</span> <span class="op">&lt;-</span> <span class="st">"pQTLs OR (protein AND quantitative AND trait AND loci) AND human [MH] AND (plasma OR Serum)"</span></span>
<span><span class="va">r</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_search.html" class="external-link">entrez_search</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"pubmed"</span>,term<span class="op">=</span><span class="va">term</span>,use_history<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">r</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">r</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">r</span>,<span class="va">web_history</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"pubmed"</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"fetch"</span>,<span class="st">"summary"</span><span class="op">)</span>,sep<span class="op">=</span><span class="st">"."</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fields</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"uid"</span>, <span class="st">"pubdate"</span>, <span class="st">"sortfirstauthor"</span>, <span class="st">"title"</span>, <span class="st">"source"</span>, <span class="st">"volume"</span>, <span class="st">"pages"</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">r</span>,<span class="va">count</span><span class="op">)</span>,<span class="fl">50</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="va">i</span><span class="op">+</span><span class="fl">49</span>, <span class="st">"records downloaded\r"</span><span class="op">)</span></span>
<span>  <span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_fetch.html" class="external-link">entrez_fetch</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"pubmed"</span>, web_history<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">r</span>,<span class="va">web_history</span><span class="op">)</span>, rettype<span class="op">=</span><span class="st">"text"</span>, retmax<span class="op">=</span><span class="fl">50</span>, retstart<span class="op">=</span><span class="va">i</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">f</span>, col.names<span class="op">=</span><span class="cn">FALSE</span>, row.names<span class="op">=</span><span class="cn">FALSE</span>, file<span class="op">=</span><span class="st">"pubmed.fetch"</span>, append<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span>  <span class="va">s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_summary.html" class="external-link">entrez_summary</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"pubmed"</span>, web_history<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">r</span>,<span class="va">web_history</span><span class="op">)</span>, rettype<span class="op">=</span><span class="st">"text"</span>, retmax<span class="op">=</span><span class="fl">50</span>, retstart<span class="op">=</span><span class="va">i</span><span class="op">)</span></span>
<span>  <span class="va">e</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/extract_from_esummary.html" class="external-link">extract_from_esummary</a></span><span class="op">(</span><span class="va">s</span>, <span class="va">fields</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">e</span><span class="op">)</span>, col.names<span class="op">=</span><span class="cn">FALSE</span>, row.names<span class="op">=</span><span class="cn">FALSE</span>, file<span class="op">=</span><span class="st">"pubmed.summary"</span>, append<span class="op">=</span><span class="cn">TRUE</span>, sep<span class="op">=</span><span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">id</span> <span class="op">&lt;-</span> <span class="fl">600807</span></span>
<span><span class="va">upload</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_post.html" class="external-link">entrez_post</a></span><span class="op">(</span>db<span class="op">=</span><span class="st">"omim"</span>, id<span class="op">=</span><span class="va">id</span><span class="op">)</span></span>
<span><span class="va">asthma_variants</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_link.html" class="external-link">entrez_link</a></span><span class="op">(</span>dbfrom<span class="op">=</span><span class="st">"omim"</span>, db<span class="op">=</span><span class="st">"clinvar"</span>, cmd<span class="op">=</span><span class="st">"neighbor_history"</span>, web_history<span class="op">=</span><span class="va">upload</span><span class="op">)</span></span>
<span><span class="va">asthma_variants</span></span>
<span><span class="va">snp_links</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_link.html" class="external-link">entrez_link</a></span><span class="op">(</span>dbfrom<span class="op">=</span><span class="st">"clinvar"</span>, db<span class="op">=</span><span class="st">"snp"</span>, web_history<span class="op">=</span><span class="va">asthma_variants</span><span class="op">$</span><span class="va">web_histories</span><span class="op">$</span><span class="va">omim_clinvar</span>, cmd<span class="op">=</span><span class="st">"neighbor_history"</span><span class="op">)</span></span>
<span><span class="va">all_links</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://docs.ropensci.org/rentrez/reference/entrez_link.html" class="external-link">entrez_link</a></span><span class="op">(</span>dbfrom<span class="op">=</span><span class="st">'pubmed'</span>, id<span class="op">=</span><span class="va">id</span>, db<span class="op">=</span><span class="st">'all'</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2" number="5">
<h2 id="gaawr2">
<span class="header-section-number">5</span> gaawr2<a class="anchor" aria-label="anchor" href="#gaawr2"></a>
</h2>
<p>While created as a showcase of modern package development, like other R packages it includes data examples, customized functions,
documentation and featured articles.</p>
<p>The relevant scripts are with <code>inst/scripts</code> directory in the source package. Briefly,</p>
<ul>
<li>
<code>gaawr2.R</code> creates the package in R.</li>
<li>
<code>github.sh</code> creates <code>gaawr2</code> at GitHub from the command line.</li>
<li>
<code>pkgdown.sh</code> makes a pkgdown-style package and this vignette is set to be processed with the <code>bookdown</code> package.</li>
<li>
<code>docs.sh</code> adds, commits and pushes files to GitHub.</li>
</ul>
<p>Note that for creation of the GitHub repository, an SSH key is assumed in place. In order for <code>pkgdown.sh</code> to function well, all
required files such as <code>nature-genetics.csl</code> need to be available.</p>
<p>A GitHub login is still necessary to enable web pages, so that this can be accessed as <a href="https://jinghuazhao.github.io/gaawr2/" class="uri">https://jinghuazhao.github.io/gaawr2/</a>.
Upon use <code>pkgdown</code>, an article can be seen from the menu item <code>Articles</code>.</p>
<p>We carry on adding files such as <code>NEWS.md</code> and <code>_pkgdown.yml</code>. In line with various analyses we have covered, their associate packages
are also added to the suggested list of packages in DESCRIPTION:</p>
<pre><code>Suggests:
    BLR,
    BGLR,
    biomaRt,
    bookdown,
    GMMAT,
    HardyWeinberg,
    haplo.stats,
    httr,
    httpuv,
    jsonlite,
    knitr,
    MCMCglmm,
    plumber,
    powerEQTL,
    regress,
    SAIGE,
    seqminer,
    SNPassoc</code></pre>
<p>Note that the <code><a href="https://devtools.r-lib.org/reference/document.html" class="external-link">devtools::document()</a></code> in <code>pkgdown.sh</code> automatically updates NAMESPACE and regenerates function documentation file(.Rd),
which can be picked up through <code><a href="https://pkgdown.r-lib.org/reference/build_reference.html" class="external-link">pkgdown::build_reference()</a></code>.</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-suhre20" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">Suhre, K., McCarthy, M. I. &amp; Schwenk, J. M. Genetics meets proteomics: Perspectives for large population-based studies. <em>Nat Rev Genet</em> (2020) doi:<a href="https://doi.org/10.1038/s41576-020-0268-2" class="external-link">10.1038/s41576-020-0268-2</a>.</div>
</div>
<div id="ref-malone10" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">Malone, J. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/btq099" class="external-link">Modeling sample variables with an experimental factor ontology</a>. <em>Bioinformatics</em> <strong>26</strong>, 1112–1118 (2010).</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Crafted by <a href="https://jinghuazhao.github.io" class="external-link">Jing Hua Zhao</a>. Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Provided without <strong>any warranty</strong>.</p>
</div>

    </footer>
</div>





  </body>
</html>
